{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "### Ссылка на диск с картированными чтениями"
      ],
      "metadata": {
        "id": "qzVRvY81sSUi"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |"
      ],
      "metadata": {
        "id": "d0f6QFwQsdza"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "[Bam-файлы и отчеты](https://drive.google.com/drive/folders/1athn52a93obOwfuHynDCRPRhjAPB80aO?usp=sharing)"
      ],
      "metadata": {
        "id": "QAzOZDDpshs-"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Установка FastQC"
      ],
      "metadata": {
        "id": "0qsOARWBsoYb"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FyS3CiPsrZ2-",
        "outputId": "c0ad30de-dab2-4b96-9159-e504c044a8e8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-05 18:22:30--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  3.38MB/s    in 2.9s    \n",
            "\n",
            "2023-02-05 18:22:34 (3.38 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Можно получать данные секвенировия с ENA. Например, https://www.ebi.ac.uk/ena/browser/view/SRR5836473"
      ],
      "metadata": {
        "id": "ckvk9pcbtAbV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
        "!gunzip SRR5836473_1.fastq.gz\n",
        "!./FastQC/fastqc SRR5836473_1.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vaOBChuntHyQ",
        "outputId": "ab01e0a8-49d6-4c13-b0ea-4278db64a703"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-05 18:22:42--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => ‘SRR5836473_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq. 100%[===================>]   6.38G  22.8MB/s    in 5m 14s  \n",
            "\n",
            "2023-02-05 18:27:58 (20.8 MB/s) - ‘SRR5836473_1.fastq.gz’ saved [6856386252]\n",
            "\n",
            "Started analysis of SRR5836473_1.fastq\n",
            "Approx 5% complete for SRR5836473_1.fastq\n",
            "Approx 10% complete for SRR5836473_1.fastq\n",
            "Approx 15% complete for SRR5836473_1.fastq\n",
            "Approx 20% complete for SRR5836473_1.fastq\n",
            "Approx 25% complete for SRR5836473_1.fastq\n",
            "Approx 30% complete for SRR5836473_1.fastq\n",
            "Approx 35% complete for SRR5836473_1.fastq\n",
            "Approx 40% complete for SRR5836473_1.fastq\n",
            "Approx 45% complete for SRR5836473_1.fastq\n",
            "Approx 50% complete for SRR5836473_1.fastq\n",
            "Approx 55% complete for SRR5836473_1.fastq\n",
            "Approx 60% complete for SRR5836473_1.fastq\n",
            "Approx 65% complete for SRR5836473_1.fastq\n",
            "Approx 70% complete for SRR5836473_1.fastq\n",
            "Approx 75% complete for SRR5836473_1.fastq\n",
            "Approx 80% complete for SRR5836473_1.fastq\n",
            "Approx 85% complete for SRR5836473_1.fastq\n",
            "Approx 90% complete for SRR5836473_1.fastq\n",
            "Approx 95% complete for SRR5836473_1.fastq\n",
            "Analysis complete for SRR5836473_1.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Установка Bismark с зависимостями (+bowtie2, samtools)"
      ],
      "metadata": {
        "id": "NDRwgSzEY4tx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()\n",
        "condacolab.check()"
      ],
      "metadata": {
        "id": "rhapBf0qY7LH",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "41d17841-ec7a-45bc-a9e9-a2fe9b4b09ba"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m✨🍰✨ Everything looks OK!\n",
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda bismark"
      ],
      "metadata": {
        "id": "SIu44SBxZJ-3",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a01ddbba-e043-42ab-a769-b2f47f868065"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bismark\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bismark-0.24.0             |       hdfd78af_0         6.6 MB  bioconda\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    ca-certificates-2022.12.7  |       ha878542_0         143 KB  conda-forge\n",
            "    certifi-2022.12.7          |     pyhd8ed1ab_0         147 KB  conda-forge\n",
            "    colorama-0.4.6             |     pyhd8ed1ab_0          25 KB  conda-forge\n",
            "    conda-22.11.1              |   py38h578d9bd_1         905 KB  conda-forge\n",
            "    hisat2-2.2.1               |       h87f3376_4        16.5 MB  bioconda\n",
            "    k8-0.2.5                   |       hd03093a_2         1.7 MB  bioconda\n",
            "    libhwloc-2.8.0             |       h32351e8_1         3.0 MB  conda-forge\n",
            "    minimap2-2.24              |       h7132678_1         484 KB  bioconda\n",
            "    openssl-3.0.7              |       h0b41bf4_2         2.5 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    pluggy-1.0.0               |     pyhd8ed1ab_5          16 KB  conda-forge\n",
            "    ruamel.yaml-0.17.21        |   py38h0a891b7_2         172 KB  conda-forge\n",
            "    ruamel.yaml.clib-0.2.7     |   py38h1de0b5d_1         143 KB  conda-forge\n",
            "    samtools-1.3.1             |                0         1.5 MB  bioconda\n",
            "    tbb-2021.7.0               |       h924138e_1         1.5 MB  conda-forge\n",
            "    tqdm-4.64.1                |     pyhd8ed1ab_0          82 KB  conda-forge\n",
            "    zlib-1.2.13                |       h166bdaf_4          92 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        64.6 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bismark            bioconda/noarch::bismark-0.24.0-hdfd78af_0 None\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 None\n",
            "  colorama           conda-forge/noarch::colorama-0.4.6-pyhd8ed1ab_0 None\n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-h87f3376_4 None\n",
            "  k8                 bioconda/linux-64::k8-0.2.5-hd03093a_2 None\n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.8.0-h32351e8_1 None\n",
            "  minimap2           bioconda/linux-64::minimap2-2.24-h7132678_1 None\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 None\n",
            "  pluggy             conda-forge/noarch::pluggy-1.0.0-pyhd8ed1ab_5 None\n",
            "  ruamel.yaml        conda-forge/linux-64::ruamel.yaml-0.17.21-py38h0a891b7_2 None\n",
            "  ruamel.yaml.clib   conda-forge/linux-64::ruamel.yaml.clib-0.2.7-py38h1de0b5d_1 None\n",
            "  samtools           bioconda/linux-64::samtools-1.3.1-0 None\n",
            "  tbb                conda-forge/linux-64::tbb-2021.7.0-h924138e_1 None\n",
            "  tqdm               conda-forge/noarch::tqdm-4.64.1-pyhd8ed1ab_0 None\n",
            "  zlib               conda-forge/linux-64::zlib-1.2.13-h166bdaf_4 None\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2022.9.24-ha878542_0 --> 2022.12.7-ha878542_0 None\n",
            "  certifi                            2022.9.24-pyhd8ed1ab_0 --> 2022.12.7-pyhd8ed1ab_0 None\n",
            "  conda                               22.9.0-py38h578d9bd_2 --> 22.11.1-py38h578d9bd_1 None\n",
            "  openssl                                  3.0.7-h0b41bf4_1 --> 3.0.7-h0b41bf4_2 None\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "samtools-1.3.1       | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.56it/s]\n",
            "tqdm-4.64.1          | 82 KB     | : 100% 1.0/1 [00:00<00:00,  9.84it/s]\n",
            "colorama-0.4.6       | 25 KB     | : 100% 1.0/1 [00:00<00:00, 15.02it/s]\n",
            "zlib-1.2.13          | 92 KB     | : 100% 1.0/1 [00:00<00:00, 14.49it/s]\n",
            "conda-22.11.1        | 905 KB    | : 100% 1.0/1 [00:00<00:00,  3.73it/s]\n",
            "hisat2-2.2.1         | 16.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.91s/it]\n",
            "ruamel.yaml.clib-0.2 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 16.40it/s]\n",
            "libhwloc-2.8.0       | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.36it/s]\n",
            "perl-5.32.1          | 14.4 MB   | : 100% 1.0/1 [00:06<00:00,  6.88s/it]\n",
            "pluggy-1.0.0         | 16 KB     | : 100% 1.0/1 [00:00<00:00, 15.04it/s]\n",
            "k8-0.2.5             | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  1.38it/s]\n",
            "ruamel.yaml-0.17.21  | 172 KB    | : 100% 1.0/1 [00:00<00:00,  5.90it/s]\n",
            "ca-certificates-2022 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 15.69it/s]\n",
            "certifi-2022.12.7    | 147 KB    | : 100% 1.0/1 [00:00<00:00, 16.59it/s]\n",
            "minimap2-2.24        | 484 KB    | : 100% 1.0/1 [00:00<00:00,  4.01it/s]\n",
            "tbb-2021.7.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  6.62it/s]\n",
            "bismark-0.24.0       | 6.6 MB    | : 100% 1.0/1 [00:02<00:00,  2.85s/it]               \n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:03<00:00,  3.88s/it]               \n",
            "openssl-3.0.7        | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  5.72it/s]\n",
            "Preparing transaction: / \b\b- \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Retrieving notices: ...working... done\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Можно загрузить файл с google drive напрямую:"
      ],
      "metadata": {
        "id": "SIxQTCBtZXiH"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# bam файлы\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n",
        "\n",
        "!cp /content/drive/MyDrive/SRR5836473_1_bismark_bt2_pe.bam /content/SRR5836473_1_bismark_bt2_pe.bam\n",
        "!cp /content/drive/MyDrive/SRR3824222_1_bismark_bt2_pe.bam /content/SRR3824222_1_bismark_bt2_pe.bam\n",
        "!cp /content/drive/MyDrive/SRR5836475_1_bismark_bt2_pe.bam /content/SRR5836475_1_bismark_bt2_pe.bam\n",
        "\n",
        "#!gdown --id 1ER0Jw61jF8291bjfMMRACQ-pFadDhHVq\n",
        "#!gdown --id 1UMTOsaLhfps-ZNccNC4Q_5Cw5FMtUTYN\n",
        "#!gdown --id 10hV1KW2Uvh8Nm_jxoBJDEpW14z3s8ZG7"
      ],
      "metadata": {
        "id": "m8uF8cGFZYVV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "c663f65e-246a-486d-a880-279c84753fc1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# отчет о выравниваниях тоже нужен\n",
        "!gdown --id 1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B\n",
        "!gdown --id 1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg\n",
        "!gdown --id 14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40"
      ],
      "metadata": {
        "id": "yVtxXy22ZbXb",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "eccdb330-d3e4-4d7f-e4a7-982aa688f503"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.8/dist-packages/gdown/cli.py:127: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B\n",
            "To: /content/SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 1.97MB/s]\n",
            "/usr/local/lib/python3.8/dist-packages/gdown/cli.py:127: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg\n",
            "To: /content/SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 2.53MB/s]\n",
            "/usr/local/lib/python3.8/dist-packages/gdown/cli.py:127: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40\n",
            "To: /content/SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "100% 1.89k/1.89k [00:00<00:00, 728kB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -lh"
      ],
      "metadata": {
        "id": "75ZDVNdvb8dr",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "87433fff-9bd5-467f-dad4-648237623f1b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 3.0G\n",
            "-rw-r--r-- 1 root root  17K Feb  7 00:52 condacolab_install.log\n",
            "drwx------ 5 root root 4.0K Feb  7 01:01 drive\n",
            "drwxr-xr-x 1 root root 4.0K Feb  3 14:34 sample_data\n",
            "-rw------- 1 root root 1.5G Feb  7 01:01 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.9K Feb  7 01:02 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw------- 1 root root 636M Feb  7 01:01 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.9K Feb  7 01:02 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw------- 1 root root 916M Feb  7 01:02 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.9K Feb  7 01:02 SRR5836475_1_bismark_bt2_PE_report.txt\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install libncurses5"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7GTipNYPafXT",
        "outputId": "f1afad7a-09cd-4545-ae28-1da8af28d362"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  libncurses5\n",
            "0 upgraded, 1 newly installed, 0 to remove and 27 not upgraded.\n",
            "Need to get 96.9 kB of archives.\n",
            "After this operation, 330 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 libncurses5 amd64 6.2-0ubuntu2 [96.9 kB]\n",
            "Fetched 96.9 kB in 0s (196 kB/s)\n",
            "Selecting previously unselected package libncurses5:amd64.\n",
            "(Reading database ... 129496 files and directories currently installed.)\n",
            "Preparing to unpack .../libncurses5_6.2-0ubuntu2_amd64.deb ...\n",
            "Unpacking libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Setting up libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view SRR5836473_1_bismark_bt2_pe.bam | head"
      ],
      "metadata": {
        "id": "a3tBEIRDb9r5",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "fc5acbfc-e1fd-4a9f-d240-b5d6b4183b21"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SRR5836473.18_18/1\t99\tchr11\t116830009\t0\t139M\t=\t116830059\t190\tTTTATTTTGTTATTGGATTTAGAGATTATTTGGCGGGAATCGGGTTTTTTTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCG\tFKKKKKKKKKKKKKKKK<FKKKKFKKKKKKKKKKKKKFKKKKKKFAFKKKKKKKKKKKKKKKAKFFFFKFF7FFFKFKK7AKKKKKKKKAFKKKKAFKKAFKFK,F<FAKKKAFAFFF,FKKKKKKKKKKKKFF<A77F\tNM:i:37\tMD:Z:0C1C2C0C2C2C6C0C23C0C0C0C0C1C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C2\tXM:Z:h.h..hx..h..x......x.............Z......Z...hhhhh.hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.18_18/1\t147\tchr11\t116830059\t0\t140M\t=\t116830009\t-190\tTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCGTTTAGTTTTTTTTTTGTAGTTCGAGTTGTTATTTTAGTTGAATTGTTTATT\tKKF<A,KAFA7A7,K77KF<F<A<F<7,,7,7A7,,7AFFFA<,KKKAA<<,7KAF7KKFAA<AKFKKF7<KKKKFKKKKKFFKKFFFKKKKKKKKKFKKKKKKKKFKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF\tNM:i:39\tMD:Z:0C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C3C5C0C1C1C2T0C2C0C7C0C2C9C0C3C3\tXM:Z:hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ..h.....hh.h.h...x..hxZ......hh..h.........xz...h...\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.30_30/1\t99\tchr11\t24232407\t0\t140M\t=\t24232506\t239\tGGTAAGAAGGATATATGTTTAATTATGTTTATAGTAGTTTTATTTATAATAGTTAGAAGTTGGAAAGAATTTAGATGTTTTTCAATAGAGGAATGGATATTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAG\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKFKKKKKAFKK\tNM:i:29\tMD:Z:0T10C1C3C1C0C1C6C4C2C0C13C0C5C9C0C0C5C0C0C0C4C13C0C13C5C1C12C1C2\tXM:Z:...........h.h...h.h..h......h....x..hh.............hx.....x.........hhx.....hhhh.H..x.............hh.............h.....h.h............h.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.30_30/1\t147\tchr11\t24232506\t0\t140M\t=\t24232407\t-239\tTTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAGTTATTAAAAAGAATGAATTTATGAAATTTTTAGGTAAATGGATGGATTTGGAGGGTATTATTTTGAGTGAGGTAATTTAATTATAAAAGAATTTATATG\t7AKKKKFFFAA7AKFAKKFKFAKKKKKFA7KKKFAFFAKKKKKKFFKFKKKKKKKKKAKKKKKFKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKA<AKFKF<AKKKKKKFKFKKKKKKKKKKKKKKKKKKFF\tNM:i:28\tMD:Z:0C0C13C5C1C12C1C2C27C0C4C0C10C0C7C2C2C0C12C0C0C3C1C7C1C1C1C0A0\tXM:Z:hh.............h.....h.h............h.x..h...........................hh....h...........hx.......h..h..hx............hhh...h.h.......h.h.h.h.\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.39_39/1\t83\tchr11\t105051790\t0\t13M1D127M\t=\t105051789\t-142\tTTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACACC\tKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G1\tXM:Z:.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h.....\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.39_39/1\t163\tchr11\t105051789\t0\t14M1D126M\t=\t105051790\t142\tATTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACAC\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKK<FKA<FKFFKKKKKKKKKFKFKKFK,,<FA,,7<\tNM:i:32\tMD:Z:0G0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G0\tXM:Z:h.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h....\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836473.41_41/1\t99\tchr11\t40185837\t0\t140M\t=\t40185875\t178\tAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATT\tKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7<77FF\tNM:i:28\tMD:Z:6G0A9C7C9C6C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C1\tXM:Z:.................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.41_41/1\t147\tchr11\t40185875\t0\t140M\t=\t40185837\t-178\tTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTTGAAGTTTTTATTATATAGATTTTTTATTTTTTTAGTG\tAKKKKKAAFFFKKKKKKKKKKKKKFFKKKKKKKKKKFAFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:4C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C8C4C3A4C3C1C2C6T0\tXM:Z:....h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h........h....h........h...h.h..h.......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.46_46/1\t83\tchr11\t22771774\t8\t140M\t=\t22771594\t-320\tCAAATTAAAAAAAAAAAAATAAAAATATCACTTTTTACAAATAATATAATAATATATATAAATAACCCTAAAAATTCTATCACAAAACTCCTAAACCTAATAAACAACTTCAATAAAATAACTAAATATAAAATTAACTC\tKKKKAAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:26\tMD:Z:6G0G4G2G2G1C15G2G2G4G3G9G1G13C1C2G1G9G3G7G5G1G2G2G2G0G15\tXM:Z:......hh....h..h..h.................h..x..h....h...h.........h.h....................h.........h...x.......x.....x.h..h..h..xh...............\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.46_46/1\t163\tchr11\t22771594\t8\t140M\t=\t22771774\t320\tAGAATAAAAACAATCTACAACAAACCAATAACCAACATCAAAATAAATAATAAAAAACTAAAAACAATCCCACTAAAATCAAAAACTAAACAAAACTACCCACTTTCTCCCTACCTATTCAACATTATACTTAAAATCCT\tFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKFFKFKKKKKKFKKKKKKK<AKKKKKKFFF7<A<K\tNM:i:25\tMD:Z:1T0G6G9G7G2G11G5G0G1G1G2G2G0G2G17G0G0G4G4G0G2G28G5G2G4\tXM:Z:..h......h.........x.......x..h...........h.....hh.h.h..h..xh..h.................xhh....h....hh..x............................h.....h..h....\tXR:Z:GA\tXG:Z:GA\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools sort SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam\n",
        "!samtools sort SRR3824222_1_bismark_bt2_pe.bam -o sorted_Epiblast.bam\n",
        "!samtools sort SRR5836475_1_bismark_bt2_pe.bam -o sorted_ICM.bam"
      ],
      "metadata": {
        "id": "zRDPsh3QcO6j",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2bd09a44-a18e-4720-960a-6202b3ad4fb6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 5 files...\n",
            "[bam_sort_core] merging from 11 files...\n",
            "[bam_sort_core] merging from 7 files...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools index  sorted_8cell.bam\n",
        "!samtools index  sorted_Epiblast.bam\n",
        "!samtools index  sorted_ICM.bam"
      ],
      "metadata": {
        "id": "ONr6qaPUcn9B"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "###Сколько ридов закартировалось на регионы\n",
        "\n",
        "*   11347700-11367700\n",
        "*   40185800-40195800\n",
        "\n"
      ],
      "metadata": {
        "id": "YSCrSjhddJa1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "!samtools view sorted_Epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "!samtools view sorted_ICM.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "id": "v-K_XWAVdpqd",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4edf22a2-b7c7-4752-c807-1a3e2d9cc95e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1090\n",
            "2328\n",
            "1456\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
        "!samtools view sorted_Epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
        "!samtools view sorted_ICM.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "id": "say2OHicd8xw",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "46523949-0f03-40f9-8be0-c456db4015df"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "464\n",
            "1062\n",
            "630\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "###Дедупликация"
      ],
      "metadata": {
        "id": "FtgxrNCAeH5-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!deduplicate_bismark  --bam  --paired  -o s_8_cell  SRR5836473_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "id": "CSsgk_80eJm2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1b8e74e4-cbad-43ef-b3c5-8667f6fe6b92"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_8_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_8_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2850231\n",
            "Total number duplicated alignments removed:\t521904 (18.31%)\n",
            "Duplicated alignments were found at:\t230432 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2328327 (81.69% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!deduplicate_bismark  --bam  --paired  -o s_Epiblast  SRR3824222_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ky7TuopN9ktg",
        "outputId": "56b50360-3064-47b1-c113-17ab666059c0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_Epiblast\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_Epiblast.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7039016\n",
            "Total number duplicated alignments removed:\t205258 (2.92%)\n",
            "Duplicated alignments were found at:\t164796 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6833758 (97.08% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!deduplicate_bismark  --bam  --paired  -o s_ICM  SRR5836475_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2BGxmp_Q9iS3",
        "outputId": "21a717cb-94bb-4be6-bca0-40890c040f80"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_ICM\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_ICM.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4159998\n",
            "Total number duplicated alignments removed:\t377882 (9.08%)\n",
            "Duplicated alignments were found at:\t229833 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3782116 (90.92% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Коллинг метилирования"
      ],
      "metadata": {
        "id": "SPzXYK9euZyz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "drive.mount('/content/drive')\n",
        "!cp /content/drive/MyDrive/mouse_chr.zip /content/mouse_chr.zip\n",
        "!unzip mouse_chr.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XXM_GW27kz2L",
        "outputId": "ce9e5e2d-4d09-4b0f-e23a-d70722ded402"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n",
            "Archive:  mouse_chr.zip\n",
            "   creating: mouse_chr/\n",
            "   creating: mouse_chr/Bisulfite_Genome/\n",
            "   creating: mouse_chr/Bisulfite_Genome/CT_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/genome_mfa.CT_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.3.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.1.bt2  \n",
            "   creating: mouse_chr/Bisulfite_Genome/GA_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/genome_mfa.GA_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.3.bt2  \n",
            "  inflating: mouse_chr/chr11.fa      \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_8_cell.deduplicated.bam"
      ],
      "metadata": {
        "id": "vAA31ytHuhg3",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "981f27f4-f4a2-448e-977d-55043d397130"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_8_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_8_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_8_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_8_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2328327 lines in total\n",
            "Total number of methylation call strings processed: 4656654\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104224209\n",
            "\n",
            "Total methylated C's in CpG context:\t2101489\n",
            "Total methylated C's in CHG context:\t269204\n",
            "Total methylated C's in CHH context:\t944695\n",
            "\n",
            "Total C to T conversions in CpG context:\t2731177\n",
            "Total C to T conversions in CHG context:\t22964928\n",
            "Total C to T conversions in CHH context:\t75212716\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.2%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_8_cell.deduplicated.txt.gz CpG_OB_s_8_cell.deduplicated.txt.gz CHG_OT_s_8_cell.deduplicated.txt.gz CHG_OB_s_8_cell.deduplicated.txt.gz CHH_OT_s_8_cell.deduplicated.txt.gz CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_8_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_8_cell.deduplicated.txt.gz\t/content/CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_8_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_8_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_8_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_8_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_8_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1736778 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_Epiblast.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p5nbQxPw9drb",
        "outputId": "55560803-16ae-4077-b4e1-1fa6903751d8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_Epiblast.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_Epiblast.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_Epiblast.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_Epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_Epiblast.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "Processed lines: 4000000\n",
            "Processed lines: 4500000\n",
            "Processed lines: 5000000\n",
            "Processed lines: 5500000\n",
            "Processed lines: 6000000\n",
            "Processed lines: 6500000\n",
            "\n",
            "Processed 6833758 lines in total\n",
            "Total number of methylation call strings processed: 13667516\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t330808061\n",
            "\n",
            "Total methylated C's in CpG context:\t11544557\n",
            "Total methylated C's in CHG context:\t1572954\n",
            "Total methylated C's in CHH context:\t3695988\n",
            "\n",
            "Total C to T conversions in CpG context:\t3465254\n",
            "Total C to T conversions in CHG context:\t71995231\n",
            "Total C to T conversions in CHH context:\t238534077\n",
            "\n",
            "C methylated in CpG context:\t76.9%\n",
            "C methylated in CHG context:\t2.1%\n",
            "C methylated in CHH context:\t1.5%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_Epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_Epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_Epiblast.deduplicated.txt.gz CpG_OB_s_Epiblast.deduplicated.txt.gz CHG_OT_s_Epiblast.deduplicated.txt.gz CHG_OB_s_Epiblast.deduplicated.txt.gz CHH_OT_s_Epiblast.deduplicated.txt.gz CHH_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_Epiblast.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_Epiblast.deduplicated.txt.gz\t/content/CpG_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_Epiblast.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_Epiblast.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_Epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_Epiblast.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_Epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_Epiblast.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_Epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_Epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_Epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_Epiblast.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_Epiblast.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_Epiblast.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2249610 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_ICM.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KAKsIpY19d52",
        "outputId": "7e193250-ace5-40ae-968f-4ae0f616c5cf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_ICM.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_ICM.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_ICM.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_ICM.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "\n",
            "Processed 3782116 lines in total\n",
            "Total number of methylation call strings processed: 7564232\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t177157754\n",
            "\n",
            "Total methylated C's in CpG context:\t1914374\n",
            "Total methylated C's in CHG context:\t297421\n",
            "Total methylated C's in CHH context:\t1066806\n",
            "\n",
            "Total C to T conversions in CpG context:\t6262904\n",
            "Total C to T conversions in CHG context:\t39194423\n",
            "Total C to T conversions in CHH context:\t128421826\n",
            "\n",
            "C methylated in CpG context:\t23.4%\n",
            "C methylated in CHG context:\t0.8%\n",
            "C methylated in CHH context:\t0.8%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_ICM.deduplicated.txt.gz CpG_OB_s_ICM.deduplicated.txt.gz CHG_OT_s_ICM.deduplicated.txt.gz CHG_OB_s_ICM.deduplicated.txt.gz CHH_OT_s_ICM.deduplicated.txt.gz CHH_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_ICM.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_ICM.deduplicated.txt.gz\t/content/CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_ICM.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_ICM.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_ICM.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_ICM.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_ICM.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_ICM.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_ICM.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_ICM.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2053827 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### M-bias plot"
      ],
      "metadata": {
        "id": "uWYRMgdkn7Pq"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Переназываем файлы\n",
        "\n",
        ".deduplication_report.txt\n",
        "\n",
        ".deduplicated_splitting_report.txt\n",
        "\n",
        ".deduplicated.M-bias.txt\n",
        "\n",
        "вручную, ставя SRRnnnnnnn_1_bismark_bt2_pe в начало."
      ],
      "metadata": {
        "id": "mkePkWNLarB2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!bismark2report"
      ],
      "metadata": {
        "id": "p15l6HDVutMN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e3d2aac1-2a1a-4518-9be0-cffe002259ab"
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR3824222_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> SRR3824222_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
            "Processing splitting report SRR3824222_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> SRR3824222_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report SRR3824222_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836473_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> SRR5836473_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
            "Processing splitting report SRR5836473_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> SRR5836473_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report SRR5836473_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836475_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> SRR5836475_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
            "Processing splitting report SRR5836475_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> SRR5836475_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report SRR5836475_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Гистограммы"
      ],
      "metadata": {
        "id": "C54_p-p_wa7f"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install pyGenomeTracks\n",
        "!pip install -U ipykernel\n",
        "!apt-get install bedtools"
      ],
      "metadata": {
        "id": "TpHncRNBwlRp",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "edc7a79e-0088-4c33-948a-d4b5aa8f9ffe"
      },
      "execution_count": 51,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: pyGenomeTracks in /usr/local/lib/python3.8/site-packages (3.8)\n",
            "Requirement already satisfied: pytest in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (7.2.1)\n",
            "Requirement already satisfied: hicmatrix>=15 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (15)\n",
            "Requirement already satisfied: pysam>=0.14 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.20.0)\n",
            "Requirement already satisfied: gffutils>=0.9 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.11.1)\n",
            "Requirement already satisfied: numpy>=1.20 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (1.24.2)\n",
            "Requirement already satisfied: pybedtools>=0.8.1 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.9.0)\n",
            "Requirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (4.64.1)\n",
            "Requirement already satisfied: bx-python>=0.8.13 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.9.0)\n",
            "Requirement already satisfied: future>=0.17.0 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.18.3)\n",
            "Requirement already satisfied: pyfaidx>=0.1.3 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.7.1)\n",
            "Requirement already satisfied: pyBigWig>=0.3.16 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (0.3.18)\n",
            "Requirement already satisfied: intervaltree>=2.1.0 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (3.1.0)\n",
            "Requirement already satisfied: matplotlib<=3.6.2,>=3.1.1 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (3.6.2)\n",
            "Requirement already satisfied: argh>=0.26.2 in /usr/local/lib/python3.8/site-packages (from gffutils>=0.9->pyGenomeTracks) (0.26.2)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.8/site-packages (from gffutils>=0.9->pyGenomeTracks) (3.18.3)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.8/site-packages (from gffutils>=0.9->pyGenomeTracks) (1.16.0)\n",
            "Requirement already satisfied: argcomplete>=1.9.4 in /usr/local/lib/python3.8/site-packages (from gffutils>=0.9->pyGenomeTracks) (2.0.0)\n",
            "Requirement already satisfied: tables>=3.5.* in /usr/local/lib/python3.8/site-packages (from hicmatrix>=15->pyGenomeTracks) (3.8.0)\n",
            "Requirement already satisfied: pandas>=0.25.* in /usr/local/lib/python3.8/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.5.3)\n",
            "Requirement already satisfied: scipy>=1.2.* in /usr/local/lib/python3.8/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.10.0)\n",
            "Requirement already satisfied: cooler>=0.8.9 in /usr/local/lib/python3.8/site-packages (from hicmatrix>=15->pyGenomeTracks) (0.9.1)\n",
            "Requirement already satisfied: sortedcontainers<3.0,>=2.0 in /usr/local/lib/python3.8/site-packages (from intervaltree>=2.1.0->pyGenomeTracks) (2.4.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (23.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (9.4.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (2.8.2)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (1.4.4)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (1.0.7)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (4.38.0)\n",
            "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (3.0.9)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (0.11.0)\n",
            "Requirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.8/site-packages (from pyfaidx>=0.1.3->pyGenomeTracks) (65.5.1)\n",
            "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (1.0.0)\n",
            "Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (1.1.0)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (22.2.0)\n",
            "Requirement already satisfied: tomli>=1.0.0 in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (2.0.1)\n",
            "Requirement already satisfied: iniconfig in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (2.0.0)\n",
            "Requirement already satisfied: h5py>=2.5 in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (3.8.0)\n",
            "Requirement already satisfied: multiprocess in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.70.14)\n",
            "Requirement already satisfied: cytoolz in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.12.1)\n",
            "Requirement already satisfied: click>=7 in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (8.1.3)\n",
            "Requirement already satisfied: asciitree in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.3)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.8/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (6.0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/site-packages (from pandas>=0.25.*->hicmatrix>=15->pyGenomeTracks) (2022.7.1)\n",
            "Requirement already satisfied: blosc2~=2.0.0 in /usr/local/lib/python3.8/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (2.0.0)\n",
            "Requirement already satisfied: numexpr>=2.6.2 in /usr/local/lib/python3.8/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (2.8.4)\n",
            "Requirement already satisfied: py-cpuinfo in /usr/local/lib/python3.8/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (9.0.0)\n",
            "Requirement already satisfied: cython>=0.29.21 in /usr/local/lib/python3.8/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (0.29.33)\n",
            "Requirement already satisfied: msgpack in /usr/local/lib/python3.8/site-packages (from blosc2~=2.0.0->tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (1.0.4)\n",
            "Requirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.8/site-packages (from cytoolz->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.12.0)\n",
            "Requirement already satisfied: dill>=0.3.6 in /usr/local/lib/python3.8/site-packages (from multiprocess->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.6)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0mLooking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.8/site-packages (6.21.1)\n",
            "Requirement already satisfied: jupyter-client>=6.1.12 in /usr/local/lib/python3.8/site-packages (from ipykernel) (8.0.2)\n",
            "Requirement already satisfied: comm>=0.1.1 in /usr/local/lib/python3.8/site-packages (from ipykernel) (0.1.2)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.8/site-packages (from ipykernel) (1.5.6)\n",
            "Requirement already satisfied: tornado>=6.1 in /usr/local/lib/python3.8/site-packages (from ipykernel) (6.2)\n",
            "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.8/site-packages (from ipykernel) (0.1.6)\n",
            "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.8/site-packages (from ipykernel) (8.9.0)\n",
            "Requirement already satisfied: debugpy>=1.6.5 in /usr/local/lib/python3.8/site-packages (from ipykernel) (1.6.6)\n",
            "Requirement already satisfied: traitlets>=5.4.0 in /usr/local/lib/python3.8/site-packages (from ipykernel) (5.9.0)\n",
            "Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in /usr/local/lib/python3.8/site-packages (from ipykernel) (5.2.0)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.8/site-packages (from ipykernel) (23.0)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.8/site-packages (from ipykernel) (5.9.4)\n",
            "Requirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.8/site-packages (from ipykernel) (25.0.0)\n",
            "Requirement already satisfied: pygments>=2.4.0 in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (2.14.0)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (0.2.0)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (5.1.1)\n",
            "Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.30 in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (3.0.36)\n",
            "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (0.18.2)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (4.8.0)\n",
            "Requirement already satisfied: stack-data in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (0.6.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.8/site-packages (from ipython>=7.23.1->ipykernel) (0.7.5)\n",
            "Requirement already satisfied: importlib-metadata>=4.8.3 in /usr/local/lib/python3.8/site-packages (from jupyter-client>=6.1.12->ipykernel) (6.0.0)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/site-packages (from jupyter-client>=6.1.12->ipykernel) (2.8.2)\n",
            "Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.8/site-packages (from jupyter-core!=5.0.*,>=4.12->ipykernel) (3.0.0)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.8/site-packages (from importlib-metadata>=4.8.3->jupyter-client>=6.1.12->ipykernel) (3.12.1)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.8/site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel) (0.8.3)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.8/site-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.8/site-packages (from prompt-toolkit<3.1.0,>=3.0.30->ipython>=7.23.1->ipykernel) (0.2.6)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/site-packages (from python-dateutil>=2.8.2->jupyter-client>=6.1.12->ipykernel) (1.16.0)\n",
            "Requirement already satisfied: asttokens>=2.1.0 in /usr/local/lib/python3.8/site-packages (from stack-data->ipython>=7.23.1->ipykernel) (2.2.1)\n",
            "Requirement already satisfied: executing>=1.2.0 in /usr/local/lib/python3.8/site-packages (from stack-data->ipython>=7.23.1->ipykernel) (1.2.0)\n",
            "Requirement already satisfied: pure-eval in /usr/local/lib/python3.8/site-packages (from stack-data->ipython>=7.23.1->ipykernel) (0.2.2)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bedtools is already the newest version (2.27.1+dfsg-4ubuntu1).\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 27 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install conda=22.9.0\n",
        "!conda install -c bioconda ucsc-fetchchromsizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F3HQc5cgAmsK",
        "outputId": "522eb18b-4ae5-475a-ed09-b1c0c3e12c92"
      },
      "execution_count": 35,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Collecting package metadata (repodata.json): | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - conda=22.9.0\n",
            "\n",
            "\n",
            "The following packages will be DOWNGRADED:\n",
            "\n",
            "  conda                              22.11.1-py38h578d9bd_1 --> 22.9.0-py38h578d9bd_2 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-fetchchromsizes\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
            "    libpng-1.6.39              |       h753d276_0         276 KB  conda-forge\n",
            "    mysql-connector-c-6.1.6    |                2         9.6 MB  bioconda\n",
            "    ucsc-fetchchromsizes-357   |       h5eb252a_2           5 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        10.2 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2 None\n",
            "  libpng             conda-forge/linux-64::libpng-1.6.39-h753d276_0 None\n",
            "  mysql-connector-c  bioconda/linux-64::mysql-connector-c-6.1.6-2 None\n",
            "  ucsc-fetchchromsi~ bioconda/linux-64::ucsc-fetchchromsizes-357-h5eb252a_2 None\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  conda                               22.9.0-py38h578d9bd_2 --> 22.11.1-py38h578d9bd_1 None\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "mysql-connector-c-6. | 9.6 MB    | : 100% 1.0/1 [00:02<00:00,  2.85s/it]\n",
            "ucsc-fetchchromsizes | 5 KB      | : 100% 1.0/1 [00:00<00:00, 14.48it/s]\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  6.94it/s]\n",
            "libpng-1.6.39        | 276 KB    | : 100% 1.0/1 [00:00<00:00, 13.63it/s]\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\bdone\n",
            "Retrieving notices: ...working... done\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! fetchChromSizes mm10 > m.chrom.sizes"
      ],
      "metadata": {
        "id": "8l2di2UgxJHT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "78e54ca2-e7f9-444b-8bab-80f551c59875"
      },
      "execution_count": 36,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO: trying WGET /usr/bin/wget for database mm10\n",
            "url: http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda ucsc-bedgraphtobigwig"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fkSAQi3bI3QY",
        "outputId": "1efff524-f8ad-425e-91a5-81edea058906"
      },
      "execution_count": 50,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-bedgraphtobigwig\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ucsc-bedgraphtobigwig-357  |                0         972 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         972 KB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  ucsc-bedgraphtobi~ bioconda/linux-64::ucsc-bedgraphtobigwig-357-0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "                                                                        \n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
        "!sudo apt-get install /content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0LDfrQ0-DZ5Q",
        "outputId": "4eb73044-184b-4e2d-8d97-68b774dab205"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-07 09:10:51--  http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
            "Resolving archive.ubuntu.com (archive.ubuntu.com)... 185.125.190.36, 185.125.190.39, 91.189.91.39, ...\n",
            "Connecting to archive.ubuntu.com (archive.ubuntu.com)|185.125.190.36|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1088952 (1.0M) [application/x-debian-package]\n",
            "Saving to: ‘libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb.6’\n",
            "\n",
            "libssl1.0.0_1.0.2n- 100%[===================>]   1.04M  1.71MB/s    in 0.6s    \n",
            "\n",
            "2023-02-07 09:10:52 (1.71 MB/s) - ‘libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb.6’ saved [1088952/1088952]\n",
            "\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "Note, selecting 'libssl1.0.0' instead of '/content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb'\n",
            "libssl1.0.0 is already the newest version (1.0.2n-1ubuntu5.10).\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'sudo apt autoremove' to remove it.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 27 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip s_8_cell.deduplicated.bedGraph.gz\n",
        "!bedGraphToBigWig  s_8_cell.deduplicated.bedGraph m.chrom.sizes cell8_methylation.bigWig\n",
        "!gunzip s_Epiblast.deduplicated.bedGraph.gz\n",
        "!bedGraphToBigWig  s_Epiblast.deduplicated.bedGraph m.chrom.sizes epiblast_methylation.bigWig\n",
        "!gunzip s_ICM.deduplicated.bedGraph.gz\n",
        "!bedGraphToBigWig  s_ICM.deduplicated.bedGraph m.chrom.sizes ICM_methylation.bigWig"
      ],
      "metadata": {
        "id": "8hd48jvpxNen"
      },
      "execution_count": 54,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g m.chromSizes  >  s_8_cell.cover.bedGraph\n",
        "!bedtools genomecov -bg -ibam s_Epiblast.deduplicated.bam -g m.chromSizes  >  s_Epiblast.cover.bedGraph\n",
        "!bedtools genomecov -bg -ibam s_ICM.deduplicated.bam -g m.chromSizes  >  s_ICM.cover.bedGraph"
      ],
      "metadata": {
        "id": "-PjuzO0Pyky4",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2a2895a7-48bd-463d-b6cf-39174857e929"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n",
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n",
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bedGraphToBigWig   s_8_cell.cover.bedGraph m.chrom.sizes cell8_coverage.bigWig\n",
        "!bedGraphToBigWig   s_Epiblast.cover.bedGraph m.chrom.sizes epiblast_coverage.bigWig\n",
        "!bedGraphToBigWig   s_ICM.cover.bedGraph m.chrom.sizes ICM_coverage.bigWig"
      ],
      "metadata": {
        "id": "mjoPMBuPy5ku"
      },
      "execution_count": 55,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import seaborn as sns\n",
        "import matplotlib.pyplot as plt"
      ],
      "metadata": {
        "id": "Nv3SXZ4Odvq6"
      },
      "execution_count": 60,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip s_8_cell.deduplicated.bismark.cov.gz\n",
        "SRR5836473 = pd.read_csv('s_8_cell.deduplicated.bismark.cov', \n",
        "                         sep='\\t', names=['chr', 'start','stop', 'Methylation %', 'Met', 'Nonmet'])\n",
        "!gunzip s_Epiblast.deduplicated.bismark.cov.gz\n",
        "SRR3824222 = pd.read_csv('s_Epiblast.deduplicated.bismark.cov', \n",
        "                         sep='\\t', names=['chr', 'start','stop', 'Methylation %', 'Met', 'Nonmet'])\n",
        "!gunzip s_ICM.deduplicated.bismark.cov.gz\n",
        "SRR5836475 = pd.read_csv('s_ICM.deduplicated.bismark.cov', \n",
        "                         sep='\\t', names=['chr', 'start','stop', 'Methylation %', 'Met', 'Nonmet'])"
      ],
      "metadata": {
        "id": "jRMg53fzdM2o"
      },
      "execution_count": 62,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "sns.histplot(data=SRR5836473, x=\"Methylation %\", bins=20, kde=True, kde_kws={'bw_adjust': 2})\n",
        "plt.title(\"8 cell\")\n",
        "plt.savefig('8_cell.png')"
      ],
      "metadata": {
        "id": "QnQVrU_Ad08p",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "outputId": "280fa9f9-c188-4ff8-fe82-3e76aaa9d42c"
      },
      "execution_count": 63,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZcAAAEWCAYAAACqitpwAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxV1b338c8vMyGBkBDmQFAiiFpUouJQa6Ui2l6xg9Njr9Tacp+qta23g23vfezkbfvctlY7cC+tVuzt41A7iK0VqVqHtiBRUZyQMCaRISQQIBAy/Z4/9kJP40kIcIYkfN+v13ll799ee691OOH8svdae21zd0RERBIpI90NEBGRgUfJRUREEk7JRUREEk7JRUREEk7JRUREEk7JRUREEk7JRWSAMLOPmdkzMetuZpPS2SY5cim5iKSAmZWb2cNmtt3MNpvZj80sK93tEkkWJReR1PgpsBUYDZwIvAe4Nq0tEkkiJReR1JgI3O/uLe6+GXgEOK67wmY2x8xWmNlOM1tjZrNDfKiZ3WFmm8yszsy+ZWaZKXoPIr2m5CKSGj8ELjezfDMbC1xAlGDewcxOBe4GvgAUAWcD68Pmu4B2YBJwEjAL+EQyGy5yKJRcRFLjKaIzlZ1ALVAF/L6bstcAd7r7EnfvdPc6d3/dzEYCFwKfdfdmd98K3ApcnoL2ixwUJReRJDOzDKKzlN8Cg4HhwDDgu93sUgasiROfAGQDm8xsh5ntAP4bGJHwRoscJiUXkeQrBsYDP3b3fe7eAPyC6Cwknhrg6G7i+4Dh7l4UXkPcvdu+G5F0UXIRSTJ33wasAz5lZllmVgTMBV7qZpc7gKvNbKaZZZjZWDOb4u6bgEeB75vZkLDtaDN7T2reiUjvKbmIpMaHgNlAPVANtAGfi1fQ3Z8FribqT2kCniS6JAZwFZADvApsBx4gGt4s0qeYHhYmIiKJpjMXERFJOCUXERFJOCUXERFJOCUXERFJOM3KGgwfPtzLy8vT3QwRkX7lueee2+bupV3jSi5BeXk5VVVV6W6GiEi/YmYb4sWTelnMzD5nZq+Y2ctmdo+Z5ZnZRDNbZmbVZnafmeWEsrlhvTpsL485zpdDfJWZnR8Tnx1i1WZ2U0w8bh0iIpIaSUsuYebXG4BKdz8eyCSaYO+7wK3uPonoJrBrwi7XANtD/NZQDjObGvY7jugmtJ+aWWaYZvwnRLPLTgWuCGXpoQ4REUmBZHfoZwGDwhP38oFNwLlEdxUDLAQuDstzwjph+0wzsxC/N8zJtI7o7uZTw6va3de6eytwLzAn7NNdHSIikgJJSy7uXgd8D9hIlFSagOeAHe7eHorVAmPD8liiifkI25uAkth4l326i5f0UIeIiKRAMi+LDSM665gIjCGaanx2suo7FGY2z8yqzKyqvr4+3c0RERkwknlZ7H3AOnevd/c2omdZnAkUhctkAOOAurBcR/QcC8L2oUBDbLzLPt3FG3qo4x+4+wJ3r3T3ytLSd4ykExGRQ5TM5LIRmBEe62rATKKZXJ8APhLKzAUeDMuLwjph++Mezaq5iOjxsLlmNhGoAJ4FlgMVYWRYDlGn/6KwT3d1iIhICiSzz2UZUaf688DKUNcC4EvAjWZWTdQ/ckfY5Q6gJMRvBG4Kx3kFuJ8oMT0CXOfuHaFP5XpgMfAacH8oSw91iIhICmjK/aCystJ1E6WIyMExs+fcvbJrXHOLJUDZ+AmY2SG9ysZPOHAFIiL9jKZ/SYDamo384NFVh7TvjbMmJ7g1IiLppzMXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJOCUXERFJuKQlFzObbGYrYl47zeyzZlZsZkvMbHX4OSyUNzO73cyqzewlMzs55lhzQ/nVZjY3Jj7dzFaGfW43MwvxuHWIiEhqJC25uPsqdz/R3U8EpgN7gN8BNwGPuXsF8FhYB7gAqAivecB8iBIFcDNwGnAqcHNMspgPfDJmv9kh3l0dIiKSAqm6LDYTWOPuG4A5wMIQXwhcHJbnAHd7ZClQZGajgfOBJe7e6O7bgSXA7LBtiLsvdXcH7u5yrHh1iIhICqQquVwO3BOWR7r7prC8GRgZlscCNTH71IZYT/HaOPGe6vgHZjbPzKrMrKq+vv6g35SIiMSX9ORiZjnARcCvu24LZxyezPp7qsPdF7h7pbtXlpaWJrMZIiJHlFScuVwAPO/uW8L6lnBJi/Bza4jXAWUx+40LsZ7i4+LEe6pDRERSIBXJ5QreviQGsAjYP+JrLvBgTPyqMGpsBtAULm0tBmaZ2bDQkT8LWBy27TSzGWGU2FVdjhWvDhERSYGsZB7czAYD5wH/EhP+DnC/mV0DbAAuDfGHgQuBaqKRZVcDuHujmX0TWB7KfcPdG8PytcBdwCDgT+HVUx0iIpICSU0u7t4MlHSJNRCNHuta1oHrujnOncCdceJVwPFx4nHrEBGR1NAd+iIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBKLiIiknBJTS5mVmRmD5jZ62b2mpmdbmbFZrbEzFaHn8NCWTOz282s2sxeMrOTY44zN5RfbWZzY+LTzWxl2Od2M7MQj1uHiIikRrLPXG4DHnH3KcA04DXgJuAxd68AHgvrABcAFeE1D5gPUaIAbgZOA04Fbo5JFvOBT8bsNzvEu6tDRERSIGnJxcyGAmcDdwC4e6u77wDmAAtDsYXAxWF5DnC3R5YCRWY2GjgfWOLuje6+HVgCzA7bhrj7Und34O4ux4pXh4iIpEAyz1wmAvXAL8zsBTP7uZkNBka6+6ZQZjMwMiyPBWpi9q8NsZ7itXHi9FDHPzCzeWZWZWZV9fX1h/IeRUQkjmQmlyzgZGC+u58ENNPl8lQ44/AktqHHOtx9gbtXuntlaWlpMpshInJESWZyqQVq3X1ZWH+AKNlsCZe0CD+3hu11QFnM/uNCrKf4uDhxeqhDRERSIGnJxd03AzVmNjmEZgKvAouA/SO+5gIPhuVFwFVh1NgMoClc2loMzDKzYaEjfxawOGzbaWYzwiixq7ocK14dIiKSAllJPv6ngV+ZWQ6wFriaKKHdb2bXABuAS0PZh4ELgWpgTyiLuzea2TeB5aHcN9y9MSxfC9wFDAL+FF4A3+mmDhERSYGkJhd3XwFUxtk0M05ZB67r5jh3AnfGiVcBx8eJN8SrQ0REUkN36IuISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIpuYiISMIlNbmY2XozW2lmK8ysKsSKzWyJma0OP4eFuJnZ7WZWbWYvmdnJMceZG8qvNrO5MfHp4fjVYV/rqQ4REUmNVJy5vNfdT3T3/Y87vgl4zN0rgMfCOsAFQEV4zQPmQ5QogJuB04BTgZtjksV84JMx+80+QB0iIpIC6bgsNgdYGJYXAhfHxO/2yFKgyMxGA+cDS9y90d23A0uA2WHbEHdf6u4O3N3lWPHqEBGRFEh2cnHgUTN7zszmhdhId98UljcDI8PyWKAmZt/aEOspXhsn3lMd/8DM5plZlZlV1dfXH/SbExGR+LKSfPyz3L3OzEYAS8zs9diN7u5m5slsQE91uPsCYAFAZWVlUtshInIkSeqZi7vXhZ9bgd8R9ZlsCZe0CD+3huJ1QFnM7uNCrKf4uDhxeqhDRERSIGnJxcwGm1nh/mVgFvAysAjYP+JrLvBgWF4EXBVGjc0AmsKlrcXALDMbFjryZwGLw7adZjYjjBK7qsux4tUhIiIpkMzLYiOB34XRwVnA/3P3R8xsOXC/mV0DbAAuDeUfBi4EqoE9wNUA7t5oZt8Elody33D3xrB8LXAXMAj4U3gBfKebOkREJAWSllzcfS0wLU68AZgZJ+7Add0c607gzjjxKuD43tYhIiKpoTv0RUQk4ZRcREQk4ZRcREQk4ZRcREQk4ZRcREQk4ZRcREQk4XqVXMzszN7EREREoPdnLj/qZUxERKTnmyjN7HTgDKDUzG6M2TQEyExmw0REpP860B36OUBBKFcYE98JfCRZjRIRkf6tx+Ti7k8CT5rZXe6+IUVtEhGRg1A2fgK1NRsPad9xZeOp2Zj4r/fezi2Wa2YLgPLYfdz93IS3SEREDkptzUZ+8OiqQ9r3xlmTE9yaSG+Ty6+B/wJ+DnQkpSUiIjJg9Da5tLv7/KS2REREBozeDkV+yMyuNbPRZla8/5XUlomISL/V2zOX/U91/EJMzIGjEtscEREZCHqVXNx9YrIbIiIiA0evkouZXRUv7u53J7Y5IiIyEPS2z+WUmNe7ga8BF/VmRzPLNLMXzOwPYX2imS0zs2ozu8/MckI8N6xXh+3lMcf4coivMrPzY+KzQ6zazG6KicetQ0REUqNXycXdPx3z+iRwMtGd+73xGeC1mPXvAre6+yRgO3BNiF8DbA/xW0M5zGwqcDlwHDAb+GlIWJnAT4ALgKnAFaFsT3WIiEgKHOqU+83AAfthzGwc8H6i+2MwMwPOBR4IRRYCF4flOWGdsH1mKD8HuNfd97n7OqAaODW8qt19rbu3AvcCcw5Qh4iIpEBv+1weIhodBtGElccC9/di1x8CX+TteclKgB3u3h7Wa4GxYXksUAPg7u1m1hTKjwWWxhwzdp+aLvHTDlBH1/c1D5gHMH78+F68HRER6Y3eDkX+XsxyO7DB3Wt72sHMPgBsdffnzOycQ2xfUrn7AmABQGVlpR+guIiI9FJvhyI/aWYjiTr0AVb3YrczgYvM7EIgj2ia/tuAIjPLCmcW44C6UL4OKANqzSwLGAo0xMT3i90nXryhhzpERCQFevskykuBZ4FLgEuBZWbW45T77v5ldx/n7uVEHfKPu/uVwBO8PV3/XODBsLyIt2/W/Ego7yF+eRhNNhGoCG1ZDlSEkWE5oY5FYZ/u6hARkRTo7WWxrwKnuPtWADMrBf7M253mB+NLwL1m9i3gBeCOEL8D+KWZVQONRMkCd3/FzO4HXiW6JHedu3eEdlwPLCbqB7rT3V85QB0iIpICvU0uGfsTS9DAQYw0c/e/AH8Jy2uJRnp1LdNCdGYUb/9bgFvixB8GHo4Tj1uHiIikRm+TyyNmthi4J6xfRpwvdREREThAcjGzScBId/+CmX0IOCts+jvwq2Q3TkRE+qcDnbn8EPgygLv/FvgtgJmdELb9U1JbJyIi/dKB+k1GuvvKrsEQK09Ki0REpN87UHIp6mHboEQ2REREBo4DJZcqM/tk16CZfQJ4LjlNEhGR/u5AfS6fBX5nZlfydjKpBHKADyazYSIi0n/1mFzcfQtwhpm9Fzg+hP/o7o8nvWUiItJv9XZusSeIplQRERE5oEN9nouIiEi3lFxERCThlFwO09cfeoXSD36V1zfvpLNTj4QREQEll8M2JC+bnNEVLH5lCw+/vIn2zs50N0lEJO2UXA7T5847hrqfXs3ZFcNZU9/MIy9vJnqkjIjIkUvJJSGck8YP46xJUYJZtXlXuhskIpJWSi4JdNL4IkYNyePJ1fXsbe1Id3NERNJGySWBMsyYeewI9rV1UrWhMd3NERFJm6QlFzPLM7NnzexFM3vFzL4e4hPNbJmZVZvZfWaWE+K5Yb06bC+POdaXQ3yVmZ0fE58dYtVmdlNMPG4dqTC8IJeKkQWsrGuipU1nLyJyZErmmcs+4Fx3nwacCMw2sxnAd4Fb3X0SsB24JpS/Btge4reGcpjZVOBy4DhgNvBTM8s0s0zgJ8AFwFTgilCWHupIicoJxbR1OC/VNqWyWhGRPiNpycUju8Nqdng5cC7wQIgvBC4Oy3PCOmH7TDOzEL/X3fe5+zqgGjg1vKrdfa27twL3AnPCPt3VkRKlhblMKMlnRc0OOnTvi4gcgZLa5xLOMFYAW4ElwBpgh7u3hyK1wNiwPBaoAQjbm4CS2HiXfbqLl/RQR9f2zTOzKjOrqq+vP5y3+g7TxhWxt62DdduaE3pcEZH+IKnJxd073P1EYBzRmcaUZNZ3sNx9gbtXuntlaWlpQo89oTifwTmZvLppZ0KPKyLSH6RktJi77yCaVfl0oMjM9s/GPA6oC8t1QBlA2D4UaIiNd9mnu3hDD3WkTEaGcezoIazf1szufe0H3kFEZABJ5mixUjMrCsuDgPOA14iSzEdCsbnAg2F5UVgnbH/co1vdFwGXh9FkE4EK4FlgOVARRoblEHX6Lwr7dFdHSk0dMwQH3VQpIkecZJ65jAaeMLOXiBLBEnf/A/Al4EYzqybqH7kjlL8DKAnxG4GbANz9FeB+4FXgEeC6cLmtHbgeWEyUtO4PZemhjpQalp/DiMJcVm9VchGRI0uvHhZ2KNz9JeCkOPG1RP0vXeMtwCXdHOsW4JY48YeBh3tbRzpUjCjgr2sa2Lm3jSGDstPdHBGRlNAd+kk2aUQBANVbdx+gpIjIwKHkkmRF+TmUFuayWslFRI4gSi4pMKm0gM07W2jWqDEROUIouaTAxOGDAVjfoBsqReTIoOSSAsMLcijIzWL9tj3pboqISEoouaSAmVFeks/Gxj2aa0xEjghKLilSPnwwrR2dbGram+6miIgknZJLipQNyyfD0ESWInJEUHJJkZysDMYOG8T6BvW7iMjAp+SSQhNLBtPY3MrOvW3pboqISFIpuaRQeRiSvE5DkkVkgFNySaGiQdkMHZTNevW7iMgAp+SSQmbGxJLB1GzfS3tHZ7qbIyKSNEouKTahJJ+OTqduh4Yki8jApeSSYmOHDSLTjI2NGjUmIgOXkkuKZWdmMKYojw1KLiIygCm5pMH4knwadreyW7Mki8gApeSSBhOKoyHJujQmvVE2fgJmdkivsvET0t18OUIl7THHZlYG3A2MBBxY4O63mVkxcB9QDqwHLnX37WZmwG3AhcAe4GPu/nw41lzg38Khv+XuC0N8OnAXMIjoccefcXfvro5kvdeDNbwgh/ycTDbofhfphdqajfzg0VWHtO+NsyYnuDUivZPMM5d24F/dfSowA7jOzKYCNwGPuXsF8FhYB7gAqAivecB8gJAobgZOA04FbjazYWGf+cAnY/abHeLd1dEnmBnji/OpadwLWLqbIyKScElLLu6+af+Zh7vvAl4DxgJzgIWh2ELg4rA8B7jbI0uBIjMbDZwPLHH3xnD2sQSYHbYNcfel7u5EZ0mxx4pXR58xoTifvW0d5Iw8Kt1NERFJuJT0uZhZOXASsAwY6e6bwqbNRJfNIEo8NTG71YZYT/HaOHF6qKNru+aZWZWZVdXX1x/8GzsMZcX5AORNPDml9YqIpELSk4uZFQC/AT7r7jtjt4UzjqQ+PaunOtx9gbtXuntlaWlpMpvxDoNzsygtyGVQ+UkprVdEJBWSmlzMLJsosfzK3X8bwlvCJS3Cz60hXgeUxew+LsR6io+LE++pjj5lfEk+ueOOpVlDkkUkhdyddduaeWb1NkZc8jV27GlNeB1JSy5h9NcdwGvu/oOYTYuAuWF5LvBgTPwqi8wAmsKlrcXALDMbFjryZwGLw7adZjYj1HVVl2PFq6NPmVCcj2Vms3RtQ7qbIiJHiMbmVu6vqmXRi2+yomYHmYOL2ba7HyUX4Ezgn4FzzWxFeF0IfAc4z8xWA+8L6xANJV4LVAM/A64FcPdG4JvA8vD6RogRyvw87LMG+FOId1dHnzK6KI/OthaeeiO1/T0icmSq3b6H+6tqaNrbxswpI/jUOUez6a4bmDSiIOF1Je0+F3d/hu7H2c6MU96B67o51p3AnXHiVcDxceIN8eroa7IyMti38WWeXl2S7qaIyAC3dWcLD654k8K8LC4+cSxDBmUntT7doZ9me9c9z9ptzdTobn0RSZLmfe089NImBuVk8uGTxyU9sYCSS9rtXf88AE+v3pbmlojIQOTuLH51My1tHfzTu8YwODdpF6z+gZJLmrU31DJmaB5Pr1a/i4gk3sq6Jmoa93J2RSmlhbkpq1fJpQ94d0Upz1Rv09MpRSShdrW08Uz1NsYX53P82CEprVvJpQ84+5hSdrW082JtU7qbIiIDyDPV2+h0OHfKCKI7NlJHyaUPOHNSCWZoSLKIJEzd9r28sWU30ycMY2gKOvC7UnLpA4ryc3jXuCL1u4hIQrg7T1fXU5CbReWEYQfeIQmUXPqI91QMZ0XNDpr2tKW7KSLSz1XX72bLzn3MOKqY7Mz0fM0rufQR75k8gk6Hv7zRJ6dBE5F+orPT+fuaBooH53Ds6NR24sdScukjTiorYnhBLo++siXdTRGRfuyNLbvYvqeN048qISPFnfixlFz6iIwM47ypI/jLqq3sa+9Id3NEpB/qdOfZ9Y2UDM7h6NLBaW2LkksfMmvqKJpbO/jbGs2SLCIHr3rrbrbvaePUicUpH3rclZJLH3L60SXk52Sy5FVdGhORg+PhrGVYfnZSZjk+WEoufUhedibnTC5lyatb6OxM6gM6RWSAWbutmYbdrZxSXpzWvpb9lFz6mPOmjqR+1z5erN2R7qaISD/y7LpGhg7KZvLIwnQ3BVBy6XPOnTySzAzjUV0aE5FeyjtqOlt37aOyfBgZGek/awEllz5naH42p00sVr+LiPSKuzP0jMspzMvi2FHpu6+lq6QlFzO708y2mtnLMbFiM1tiZqvDz2EhbmZ2u5lVm9lLZnZyzD5zQ/nVZjY3Jj7dzFaGfW63MDSiuzr6k1lTR1K9dTfVW3enuyki0sf9bU0DeWOPpXLCMDL7yFkLJPfM5S5gdpfYTcBj7l4BPBbWAS4AKsJrHjAfokQB3AycBpwK3ByTLOYDn4zZb/YB6ug3Zh8/GjN46MU3090UEenjbntsNe27Gpiaxrvx40lacnH3p4DGLuE5wMKwvBC4OCZ+t0eWAkVmNho4H1ji7o3uvh1YAswO24a4+1J3d+DuLseKV0e/MWpoHjMmlrDoxTeJ3p6IyDv9fU0Dz65rZOeyB8hK0xxi3Ul1a0a6+6awvBkYGZbHAjUx5WpDrKd4bZx4T3W8g5nNM7MqM6uqr+9bMxLPOXEM67Y1s7JOz3gRkfhue+wNSgtz2f3i4nQ35R3SlurCGUdS/yw/UB3uvsDdK929srS0NJlNOWgXHD+anMwMfv+CLo0daTY17eWXf1/PF379IlcsWMrIK77NH1/axNK1DWze2aKzWQFg2doGlq5t5H+/52i8vTXdzXmHrBTXt8XMRrv7pnBpa/8UwHVAWUy5cSFWB5zTJf6XEB8Xp3xPdfQrQ/OzOXfKCB5cUcdNF0whJ6tvnfJK4j23YTs/faKax16PfmWHF+QyoSQfgG2797GmfjfL1kV3YE+fMIxjRw3pM8NOJfVue2w1wwtyufK08Xwi3Y2JI9XJZREwF/hO+PlgTPx6M7uXqPO+KSSHxcB/xHTizwK+7O6NZrbTzGYAy4CrgB8doI5+57JTy3jklc38+bUtXHjC6HQ3R5Kkftc+vvXHV3lwxZsUD87hhpkVXDRtzFtTeNi1Z/KlR1fR0tbBmvrdvFjbxJ9f28oLNTs4d/IIxhQNSvM7kFRbvr6Rv61p4N/efyx52Znpbk5cSUsuZnYP0VnHcDOrJRr19R3gfjO7BtgAXBqKPwxcCFQDe4CrAUIS+SawPJT7hrvvHyRwLdGItEHAn8KLHurod86uKGXM0DzuXV6j5BKUjZ9Abc3GQ95/XNl4ajZuSGCLDs9fq7fxmXtXsLOljU+fO4lPnXM0+Tnx/1vmZWdy3JihTB09hDX1zTy1up4HnqtlxlElnFI+LO0TFUpquDvff3QVwwtyuPK0CeluTreSllzc/YpuNs2MU9aB67o5zp3AnXHiVcDxceIN8erojzIzjEsqy7j98dXUNO6hrDg/3U1Ku9qajfzg0VWHvP+NsyYnsDWHrr2jk9sfW82PnqjmqOGD+Z9PnMqUXt4AZ2ZMGlHA+OJ8Hn99K39f20Ddjr2cf9zIbhOTDByPv76VpWsb+cac4xiU0zfPWkB36Pd5l51SRoYZv1zad/7alsOzq6WNaxZWcfvj1Xzk5HE89Omzep1YYuVkZXD+cSM5d8oI6nbs5Z5na9i2e18SWix9RXtHJ9/+0+scNXwwV5w6Pt3N6ZGSSx83pmgQF54wmnuWbWT3vvZ0N0cO05s79nLJf/2dZ6q38e0PncB/XjLtsM42zIwTxg7l0spxOM6vq2rZ2LgngS0+dGXjJ2Bmh/QqG993L/ek031VNVRv3c2XLphCdh+7r6UrnUP3A9ecNZGHXnyT+5fX8PGzJqa7OXKIXq5r4pqFy9mzr4O7rj6Fd1ckbvj7iMI8Lqss48EVb/LgijrOO3YkU9J8x/bhXMLsK5cv+5Ld+9q5dckbnFpezKyp3d6+12f07dQnAJxYVkTlhGHc8cw6PQK5n3r89S1c+t9/Jysjgwc+dUZCE8t+hXnZXDJ9HGOKBrH41S0sX991ggzpz/77yTVs293KV95/bL8YvKHk0k98emYFdTv2cn9V7YELS5/h7tzxzDo+sbCKo0sL+N21ZzB5VPKet5GbncmcE8cweVQhf1vTQPH519PW0Zm0+iQ11tTv5r+fXMtF08ZwYllRupvTK0ou/cTZFcOpnDCMnzxeTUubzl76g9b2Tr7yu5V88w+vMmvqKO77lxmMGJKX9HqzMjI4f+pITikfRuGJs/n4XcvZ2dKW9HolOdydr/5uJXnZGfzbB45Nd3N6TcmlnzAzbjzvGDbvbOEXf12f7ubIAWzd1cJVdy7jnmdruO69R/PTK09O6TBhM+OMo4ez7eHb+PuaBj4y/2/U9JGOfjk4v1q2kaVrG7npgmMZUZj8P04SRcmlHzlj0nDed+xIfvT4ajY17U13c6QbT7y+lQt++DQranZw62XT+ML5U9I2TUvzyiUs/PipbGpq4QM/eoZHX9mclnbIoVlbv5tb/nn0DOgAABGoSURBVPga764YzuWnlB14hz5EyaWfufmfptLe6XzrD6+luylp5+7sbetgy84WVm/Zxcq6Jl7YuJ3l6xup2tDIytom3tiyi7rte2ne1570CR+b97XztUWvcPVdyyktzOWh68/igyeNO/COSXbmpOH84dNnMb44n3m/fI6vP/QKre3qh+nrWto6+My9K8jNzuB7l0zrd/PIaShyP1NWnM+n3zuJ7y95g1kr6phz4tgD7zRAtLZ3kjNmMs9t2M6bO/ayqamFvQfR/5STmcHIK/8v//Hwa0yfMIyTxw+jtDD3sNvV3tHJohff5P8+sorNO1v4+JkT+eLsyX1qzqcJJYN54FOn8+2HX+cXf13P36ob+I8PncD0Cf3uQa1HBHfn33//MivrmvjZVZWMTEFfXaIpufRDnzrnaJ5YtZV/+/3LnDx+2ICdFmZXSxvPb9xB1fpGnl3XyIqaHYz+5+/zTPU2hg7Kpnx4PsMLchk6KJshedkMys4kK9PIyjAc2NfWyb72Dnbva2f7nja2N7fy7Ebjrr+uZ8FTawGYUJLP9AnDqJxQzPQJw6gYUdDrvxAbm1v5/Qt1/HLpBtZta+a4MUP4yZUnMX1CcRL/VQ5dblYmX7voON5dMZx///3LfHj+3/jAu0bz2fdVMGlEYkawtXd0sr5hD9Vbd1FYeRFPvlFP8752Wjs6ae9w2js7ycrIICvTyM7MoCA3i8K8LApzsygpyKUoP5uMfjDMNtl+/vQ6fv1cLTecO4nz+sE9LfEoufRDWZkZ/PCyk3j/j57mEwureOBTp1OYl53uZh22zU0t0SWt9Y0sX7+d1zfvpNOjOdaOGzOEK0+bwHc//0m+/P2fMTj3wL+62ZkZFBB9aU0oiWIP3fAF1re288qbTTy3YTtV67fz1Bv1/Pb56IkNQ/KymFZWxFHDB1M+fDBlw/IZMiibDCNcgtvHG1t28fyG7Ty/cTudDtPKivivj57MrKmj+sWli5nHjmTGUSX815NruOOZdfzhpU2855hSPnTyWN47ZQRDevG75O7U797HG5t3s2rLLl7ftJPXN+/ijS272BcuuRXPnMfLdU0U5GWRk5lBdmYGeVmZdHQ6LW0d7NzbxoaGZto63r5cmZVhlBTkUHz+9fzP0g2cWFbE5FGFff5u9ET6zXO13PLwa7z/hNF89n3HpLs5h0zJpZ8aX5LP/CunM/cXz3Ltr57nZ1dV9qnLMD0pGz+Buk2byR5xFLmjjyF3zDHkjplCVtEoADpb97LvzdfZV/tq9HpzFWvbWngo7N+bxNKTvOxMpk8oZvqEYuadHX1RbmjYEyWbDdt5ua6J3z5fx65uptvJyczg2DFDuP7cCmYfN4qpY/rWs8t7Y3BuFv86azIfO6OcXy3byD3PbuQz964gw2DqmCFMHjmE0UPzyMvOIC87SggNza1s272P2u17Wb1lF9v3vD28eXhBLseOLuSq0ycwZdQQjhlZyIkVZfzng8t7vOHP3dnX3snOvW1sa25l26591O/eR/7kM/m3378MQG5WBsePHcq0cUVMKxvKSWXDKCse1C9uJDxY9zy7ka/8biVnTirhB5f1v36WWEou/dhZFcP59odO4Eu/eYmrf7Gcn82tpOAwv3iTYU9rO29s2c2qzTtZWddE+7k3Uj56Ep3hD9aC3CxGDc1jzNA8xhQNorQgl4yMd8U9VjKmBTEzysOZyoenRx3w7s623a28uWMvu1racZzcrExGFOYypmjQgHl4W0lBLjfMrOD6907i+Y3beWr1Np7fsJ1nquup37Xvrc8IookySwtyGTU0j9nHj2byyAKOGVXIMSMLGV7wzr6rzpZdB0wAZkZediZ52ZnRPUDhyRI33jSTjQ3NrKjZwYqaHbxYs4NfLdvAnX8NZ0WDc5g2bijTyoqYVlbEieOKGDY4J2H/LqnW0enc9uc3uP3xas6ZXMr8K6eTm9U//ljsTt/7JpKDcmllGdmZxud//RJzfvwM8z86nWNGJu8O8J50dDobG/e8dYnk9c07WbV5Fxsa97B/oFZhbhadLbs5efwwRg3NY9SQvMM+E0kGM6O0MDchHf79QUaGUVleTGX52/1F7k5bh9PS3oER/RGQyrOFsuJ8yorz+adpYwBo6+hk1eZdvFi7gxUbd/Bi7Q7+8kb9W79b44vzOWZkIY8+cDeNa1fS1lBDW0Mt3n5wM0Wn+pk/b+7Yy5d+8xJPr97GJdPHccsHTxgQf7z0vf/VctA+eNI4RhbmccO9L/CBHz3Dv5x9FP/7PUcn7Uvb3dmyM3rs7qqYJLJqyy5a2qK/LDMMyksGM3XMED540jgmjyrk2NGFlA3LJzNzNmdec+jPZJHUMDNysqzPfNFlZ0aXx44fO/Sth2Tt3tfOS7U7eLGmiZV1O1i9ZTc2ZSbDjzv/rf2G5GVRlJ/D0EHZFOVnM3TQ2694fTmpmjRzT2s7C/+2gR8/vppOh//44AlccWrZgLncp+QyQJwxaTgPf+bd3PLH1/jR49Us/Nt6Lj91PBdNG8NxY4Yc0i9sS1sHGxv3sGbrbtbU72ZNfXP0c+tumlvfHgI8vCCHyaMKufK0CUweVciUUYVUjCjs0w8yOhSH8xTMvvYEzIGiIDeLM44ezhlHD38rZplZfPWBF2hsbn3rtWNvG1t2trw12GC/wbmZUdIZlPNWwskdO4VNTXsZUZhHZhL6PKq37ua3z9dy3/IaGppbOXfKCL5+0XEDbtSnkssAMqIwj9suP4mPnVHOgqfWcscz61jw1FpKBufwrnFDGTcsn9FFeeE/DRhGR6ezq6WNpr3tbN/Tyqamvby5o4U3d+ylobn1H44/ZmgeR48o4JLKMo4uHczRpQVUjCw8Yi4dHYlTyB/uY6XTorODkoJcSuL0A7W0dbBjbxtNe9po2tvGjr2tNO2JRq3t/4Np1Ee/x+nffpzMDGPUkDzGFEV9gWOKBjGiMJfiwTmUDM6NRrUNziEvO5PcrAxyszLe+iOuvaOTPW0dNOxuZUNDMxsb9/Dapp0sW9vI2m3NZFg0am/e2UdxSnnfHLp+uAZscjGz2cBtQCbwc3f/TpqblDInjR/G/I9Op7G5lT+/toVn1zXycl009HZnS/cPHCvIzWJ06FQ/fuxQxhblUVacz9GlBUwcPvgfLrP1yy8dOWgDLaHmZWcyKjuTUXFuSmzriEatff/zHyersISsIaU0DRnBmiGlZA4pJauwBMvseZi2d7SRm5NLa5yZqIfkZVFZXszHzizn/ONG9csbIw/GgEwuZpYJ/AQ4D6gFlpvZInd/Nb0tS63iwTlcWlnGpZVvz0nUvK+dbbvfHgVkQGFeFkO6uf7cnYH2pSOSnZlBSUEuLWur4v5uuzstbZ3sbetgT2s7e1s72NvWQUen0x5eS+5ZwBe/9EXys7PIz8mkKD+b8uGDmVCcT2lh7oDpT+mNAZlcgFOBandfC2Bm9wJzgAGVXA737CEzK5uO9iNsKnbLSM9/8COt3sPVB9ttZgzKyWRQTibF3Qx7/vVTd/OVJ+86pOMPtP+PluzJ/NLBzD4CzHb3T4T1fwZOc/fru5SbB8wLq5OBQx3CNBzYdoj79ld6z0cGveeB73Df7wR3f8ejVQfqmUuvuPsCYMHhHsfMqty9MgFN6jf0no8Mes8DX7Leb98YwJ54dUDsww/GhZiIiKTAQE0uy4EKM5toZjnA5cCiNLdJROSIMSAvi7l7u5ldDywmGop8p7u/ksQqD/vSWj+k93xk0Hse+JLyfgdkh76IiKTXQL0sJiIiaaTkIiIiCafkcpjMbLaZrTKzajO7Kd3tSTQzKzOzJ8zsVTN7xcw+E+LFZrbEzFaHnwPuYexmlmlmL5jZH8L6RDNbFj7r+8JgkQHDzIrM7AEze93MXjOz0wf652xmnwu/1y+b2T1mljfQPmczu9PMtprZyzGxuJ+rRW4P7/0lMzv5UOtVcjkMMdPMXABMBa4ws6npbVXCtQP/6u5TgRnAdeE93gQ85u4VwGNhfaD5DPBazPp3gVvdfRKwHbgmLa1KntuAR9x9CjCN6L0P2M/ZzMYCNwCV7n480eCfyxl4n/NdwOwuse4+1wuAivCaB8w/1EqVXA7PW9PMuHsrsH+amQHD3Te5+/NheRfRF85Yove5MBRbCFycnhYmh5mNA94P/DysG3Au8EAoMqDes5kNBc4G7gBw91Z338EA/5yJRswOMrMsIB/YxAD7nN39KaCxS7i7z3UOcLdHlgJFZjb6UOpVcjk8Y4GamPXaEBuQzKwcOAlYBox0901h02ZgZJqalSw/BL4I7J/etgTY4e77p5UeaJ/1RKAe+EW4FPhzMxvMAP6c3b0O+B6wkSipNAHPMbA/5/26+1wT9p2m5CK9YmYFwG+Az7r7zthtHo1nHzBj2s3sA8BWd38u3W1JoSzgZGC+u58ENNPlEtgA/JyHEf2lPhEYAwzmnZePBrxkfa5KLofniJhmxsyyiRLLr9z9tyG8Zf/pcvi5NV3tS4IzgYvMbD3Rpc5zifojisLlExh4n3UtUOvuy8L6A0TJZiB/zu8D1rl7vbu3Ab8l+uwH8ue8X3efa8K+05RcDs+An2Ym9DXcAbzm7j+I2bQImBuW5wIPprptyeLuX3b3ce5eTvSZPu7uVwJPAB8JxQbae94M1JjZ/oftzCR6RMWA/ZyJLofNMLP88Hu+/z0P2M85Rnef6yLgqjBqbAbQFHP57KDoDv3DZGYXEl2f3z/NzC1pblJCmdlZwNPASt7uf/gKUb/L/cB4YANwqbt37TTs98zsHODz7v4BMzuK6EymGHgB+Ki770tn+xLJzE4kGsCQA6wFrib6A3TAfs5m9nXgMqJRkS8AnyDqYxgwn7OZ3QOcQzS1/hbgZuD3xPlcQ5L9MdHlwT3A1e5edUj1KrmIiEii6bKYiIgknJKLiIgknJKLiIgknJKLiIgknJKLiIgknJKLSBxm5mb2PzHrWWZWv3+G5B72OzEMT9+//jUz+/xB1FseO3ttD2X+V8x6pZnd3ts6ejhurpk9EmYIvjYmvuBwZseVI5OSi0h8zcDxZjYorJ9H7+5UPhG48IClDk858FZycfcqd78hAcc9H3gGeBfwzwBmNg3I3D95qUhvKbmIdO9hopmRAa4A7tm/wcwGh+dkPBsmepwTZmn4BnCZma0ws8tC8alm9hczW2tmN4T9v2Fmn4053i0WnpUTEys3s6fN7PnwOiNs+g7w7lDH58zsHHv7mTPFZvb78CyOpWb2rhD/WmjvP7SjizaimYGzAQuxbwL/fqj/gHLkUnIR6d69wOVmlkf01/yymG1fJZoW5lTgvcB/En0p/x/gPnc/0d3vC2WnEJ0VnArcHOZquxO4CsDMMoimmfkf/tFW4Dx3P5noLvL9l75uAp4OddzaZZ+vAy+4+7uIZlK4O2ZbvHbEWkJ0VrQUuN3MLgKed/c3e/5nEnmnrAMXETkyuftL4TEDVxCdxcSaRTS55f7+lDyiqTTi+WOYPmSfmW0lmu58vZk1mNlJRNOdv+DuDWZWGLNfNvDjMC1LB3BML5p9FvDh0P7HzazEzIZ01w6iCSv3v992wuW2kHgWA3PM7Afhvd3t7gNq7jxJHiUXkZ4tInrmxzlEz3TZz4APu/uq2MJmdlqcY8TOS9XB2//vfg58DBhFdCbT1eeI5oKaRnSVoeWgW9+7dsRzLdFZzwyi55xcBjzOAJuYVZJHl8VEenYn8HV3X9klvhj4dJjoj3AGArALKKR3fkc0QeAp4XhdDQU2uXsnUQd7Zi/qeBq4MrTpHGBb1+fvHEh4zskHiJJLPtGEpQ4M6mk/kVhKLiI9cPdad483zPebRJetXjKzV8I6RNO1T+3Sod/dsVtD+fvdvSNOkZ8Cc83sRaL+kuYQfwnoMLMXzexzXfb5GjDdzF4i6vify8H7P8AtIaktBt5NNCv2Lw/hWHKE0qzIImkSOvKfBy5x99Xpbo9IIunMRSQNzGwqUA08psQiA5HOXEREJOF05iIiIgmn5CIiIgmn5CIiIgmn5CIiIgmn5CIiIgn3/wHy72ZPk49sowAAAABJRU5ErkJggg==\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sns.histplot(data=SRR3824222, x=\"Methylation %\", bins=20, kde=True, kde_kws={'bw_adjust': 2})\n",
        "plt.title(\"Epiblast\")\n",
        "plt.savefig('Epiblast.png')"
      ],
      "metadata": {
        "id": "ptjYn1-4d2PR",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "outputId": "b99ea6ac-f33c-4faa-bdf5-cff20cd1c1a0"
      },
      "execution_count": 64,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxddZ3/8dfnZk+arkm6r3SByk5BkEUWRUCljguLCjoqzPwUHUGcwXF+ijjOT0fHbURHREZAZFMGkalUNgWVlpYCla60pfuSNGmzL3f5/P64J3hJkzZpc+5N7nk/H4/7SM5y7/kcTjnvnO/33O8xd0dERKIrlusCREQktxQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCkX4ys9+a2UeC3z9qZn88yLq/N7NPZK86kcOnIJDIMbPNZtZuZi0Zrx8c6n3ufrG735mNGgHM7GYz+3m2tifRVZjrAkRy5N3u/kSuixAZCnRFIBIImnv+ZGY/MLNGM1trZhdkLO/Z3GN9rdvjc48ys6fMrN7M9prZPWY2OmP5P5nZDjNrNrN1ZnaBmV0E/DNweXDF8nJoOy6RpyAQeaM3AxuBKuDLwENmNvYI1zXg/wGTgGOAqcDNAGY2D7gOONXdK4F3AJvd/THg34D73X2Eu58wOLsncqBhGQRmdoeZ1ZrZK/1c/zIzW21mq8zsF2HXJ8PCw2a2P+N1TTC/Fviuu8fd/X5gHfDOPj6jX+u6+wZ3f9zdO929Dvg28NZgcRIoAeabWZG7b3b3jYO4nyKHNCyDAPgZcFF/VjSzOcAXgDPd/U3AZ0OsS4aP97j76IzXT4L5O/yNIzFuIf2XfG/6ta6ZjTez+4Lmnybg56SvInD3DaT/Td4M1Abr9bU9kVAMyyBw92eAhsx5QTvsY2b2gpk9a2ZHB4uuAW51933Be2uzXK4ML5PNzDKmpwE7j3DdfwMcOM7dRwIfJt1cBIC7/8LdzwKmB+t9o3vR4e2CyMAMyyDow23Ap939FOBG4IfB/LnA3KATcEnQCSfSlxrgM2ZWZGYfIN2mv+gI160EWoBGM5sMfL57gZnNM7PzzawE6ADagVSweA8ww8zy6f9TGYLy4vZRMxsBvAV4MOMPtJLgZyEwBzgXmAI8Y2bHufv+bNcpQ8pvzCyZMf048GtgKel/L3tJn4jf7+71fXxGf9f9CnAX0AhsAO4Grg+WlQBfJx0iceDPwLXBsgdJXz3Um9lr7n7yYeynyCHZcH0wjZnNAB5192PNbCSwzt0n9rLefwFL3f2/g+kngZvcfVk265Whz8w+CnwiaKYRiYy8uOR09ybgteDyHEvrvt3uYdJXA5hZFemmok25qFNEZCgalkFgZvcCzwHzzGy7mX0c+BDw8eCLN6uAhcHqi0lfWq8GngY+f5BLfRGRyBm2TUMiIjI4huUVgYiIDJ5hd9dQVVWVz5gxI9dliIgMKy+88MJed6/ubdmwC4IZM2awfPnyXJchIjKsmNmWvpapaUhEJOIUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIlkyddp0zOywX1OnTQ+lrmE3xISIyHC1fdtWvv27dYf9/hsunDeI1fyVrghERCJOQSAiEnEKAhGRiFMQiIhEXGhBYGZ3mFmtmb3Sx3Izs++b2QYzW2lmJ4dVi4iI9C3MK4KfARcdZPnFwJzgdS3woxBrERGRPoQWBO7+DNBwkFUWAnd52hJgtJlNDKseERHpXS77CCYD2zKmtwfzDmBm15rZcjNbXldXl5XiRESiYlh0Frv7be6+wN0XVFf3+uxlERE5TLkMgh3A1IzpKcE8ERHJolwGwSPA1cHdQ6cDje6+K4f1iIhEUmhjDZnZvcC5QJWZbQe+DBQBuPt/AYuAS4ANQBvwt2HVIiIifQstCNz9ykMsd+BTYW1fRET6Z1h0FouISHgUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIhGnIBARiTgFgYhIxCkIREQiTkEgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGIUxCIiEScgkBEJOIUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnGhBoGZXWRm68xsg5nd1MvyaWb2tJm9aGYrzeySMOsREZEDhRYEZlYA3ApcDMwHrjSz+T1W+xfgAXc/CbgC+GFY9YiISO/CvCI4Ddjg7pvcvQu4D1jYYx0HRga/jwJ2hliPiIj0IswgmAxsy5jeHszLdDPwYTPbDiwCPt3bB5nZtWa23MyW19XVhVGriEhk5bqz+ErgZ+4+BbgEuNvMDqjJ3W9z9wXuvqC6ujrrRYqI5LMwg2AHMDVjekowL9PHgQcA3P05oBSoCrEmERHpIcwgWAbMMbOZZlZMujP4kR7rbAUuADCzY0gHgdp+RESyKLQgcPcEcB2wGFhD+u6gVWZ2i5ldGqz2OeAaM3sZuBf4qLt7WDWJiMiBCsP8cHdfRLoTOHPelzJ+Xw2cGWYNIiJycLnuLBYRkRxTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGIUxCIiEScgkBEJOIUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIhGnIBARiTgFgYhIxCkIREQiTkEgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIi7UIDCzi8xsnZltMLOb+ljnMjNbbWarzOwXYdYjIiIH6lcQmNmZ/ZnXY3kBcCtwMTAfuNLM5vdYZw7wBeBMd38T8Nl+1i0iIoOkv1cE/9nPeZlOAza4+yZ37wLuAxb2WOca4FZ33wfg7rX9rEdERAZJ4cEWmtkZwFuAajO7IWPRSKDgEJ89GdiWMb0deHOPdeYG2/lT8Hk3u/tjvdRxLXAtwLRp0w6xWRERGYhDXREUAyNIB0ZlxqsJeP8gbL8QmAOcC1wJ/MTMRvdcyd1vc/cF7r6gurp6EDYrIiLdDnpF4O5/AP5gZj9z9y0D/OwdwNSM6SnBvEzbgaXuHgdeM7P1pINh2QC3JSIih+mgQZChxMxuA2Zkvsfdzz/Ie5YBc8xsJukAuAL4YI91HiZ9JfDfZlZFuqloUz9rEhGJBHfntfpWrLgslM/vbxA8CPwXcDuQ7M8b3D1hZtcBi0m3/9/h7qvM7BZgubs/Eiy70MxWB5/7eXevH+hOiIjkq+aOOE+trWVzfRuVJ10Syjb6GwQJd//RQD/c3RcBi3rM+1LG7w7cELxERCSDu/PbV3azt6WTs2dX8fNvPhzKdvp7++hvzOyTZjbRzMZ2v0KpSEREANja0Mauxg7Oml3FydPHQKpfDTID1t8rgo8EPz+fMc+BWYNbjoiIQPpqYMmmBipLC5k/aWSo2+pXELj7zFCrEBGRN9jT1Mnupg7Om1dNYSzcYeH6FQRmdnVv8939rsEtR0READbWtRAzmDu+MvRt9bdp6NSM30uBC4AVgIJARCQEG+tamDymjNKiQw3icOT62zT06czp4Nu/94VSkYhIxDW0drGvLc4JUw4YaCEUh9vw1Aqo30BEJASb6loAmFVdkZXt9beP4Dek7xKC9JfDjgEeCKsoEZEo21zfRnVlCZWlRVnZXn/7CL6V8XsC2OLu20OoR0Qk0pIpZ3dTB8dNHpW1bfaraSgYfG4t6ZFHxwBdYRYlIhJVtc0dJFPOpNGlWdtmf59QdhnwPPAB4DJgqZkNxjDUIiKSYef+DgAmjQpngLne9Ldp6IvAqd1PEDOzauAJ4JdhFSYiEkU797czqqyIipL+np6PXH/vGor1eIxk/QDeKyIi/eDu7GxsZ/Lo7F0NQP+vCB4zs8XAvcH05fQYVVRERI7MvrY4HfEUE7PYPwCHfmbxbGC8u3/ezN4LnBUseg64J+ziRESiZE9Tun9g4sghFATAd4EvALj7Q8BDAGZ2XLDs3aFWJyISIbXNnRTGjDHlxVnd7qHa+ce7+196zgzmzQilIhGRiKpt7qC6soRYzLK63UMFwcEGushub4aISB5zd+qaO6mpLMn6tg8VBMvN7JqeM83sE8AL4ZQkIhI9+9vixJNOdQ6C4FB9BJ8F/sfMPsRfT/wLgGLgb8IsTEQkSvY0pzuKayqz21EMhwgCd98DvMXMzgOODWb/r7s/FXplIiIRUtfcSUHMGFuR3Y5i6P/zCJ4Gng65FhGRyKpt7qRqRDEFWe4oBn07WEQk59yd+pYuqkZkv38AFAQiIjnX1pWkPZ5UEIiIRFV9a3pk/3E56B8ABYGISM7tbekEYNwIBYGISCTVt3RRXlxAeXH2hp7OpCAQEcmx+tbOnDULgYJARCSnuu8YGpejjmJQEIiI5FRje5xEyqnKUf8AhBwEZnaRma0zsw1mdtNB1nufmbmZLQizHhGRoeavdwzl4RWBmRUAtwIXA/OBK81sfi/rVQL/ACwNqxYRkaEq13cMQbhXBKcBG9x9k7t3AfcBC3tZ76vAN4COEGsRERmS6lu6GFVWRFFB7lrqw9zyZGBbxvT2YN7rzOxkYKq7/2+IdYiIDFn1LV05vWMIcthZbGYx4NvA5/qx7rVmttzMltfV1YVfnIhIFiRSKfa1526MoW5hBsEOYGrG9JRgXrdK0kNb/97MNgOnA4/01mHs7re5+wJ3X1BdXR1iySIi2bOvNY57bvsHINwgWAbMMbOZZlYMXAE80r3Q3RvdvcrdZ7j7DGAJcKm7Lw+xJhGRIaO+u6M4X5uG3D0BXAcsBtYAD7j7KjO7xcwuDWu7IiLDxd7WLgrMGF2e2yAIdWALd18ELOox70t9rHtumLWIiAw1e1s6GVNRlJOH0WTSN4tFRHIk10NLdFMQiIjkQGc8SUtngqoc9w+AgkBEJCdeH1pCVwQiItE0FIaW6KYgEBHJgfrWLooLYlSW5OZhNJkUBCIiOZDuKC7GLLd3DIGCQEQk69ydvS25fSpZJgWBiEiWtXYl6Uykcj7GUDcFgYhIltUPoY5iUBCIiGRdfUvun0qWSUEgIpJle1s7KS8uoKy4INelAAoCEZGsq2/J/TMIMikIRESyKOVOfWvXkOkfAAWBiEhWNbbHSaZ8yNw6CgoCEZGs6u4oVtOQiEhEdY8xNFZXBCIi0VTf2sWosiKKCobO6XfoVCIiEgH1LZ1UDaGOYlAQiIhkjRWVsq8tPqT6B0BBICKSNcU1MwCoqVQQ5MzUadMxs8N+TZ02Pde7ICLDWFHNUQBUD7EgyP0TEbJo+7atfPt36w77/TdcOG8QqxGRqCkeP4vSohgjhsDDaDJF6opARCSXiscfRXVlyZB4GE0mBYGISBbEkymKq6ZTU1ma61IOoCAQEcmCV/e0YIVFVA+xO4ZAQSAikhWv7GwEht4dQ6AgEBHJipe27SfV0cLo8qJcl3IABYGISBa8tHU/nbvWD7mOYlAQiIiErq0rwbo9zXTuWp/rUnqlIBARCdlftjeSTDldOw//e0xhUhCIiITspW37AeiMYhCY2UVmts7MNpjZTb0sv8HMVpvZSjN70sw0hoOI5J2Xtu1n2thyUu1NuS6lV6EFgZkVALcCFwPzgSvNbH6P1V4EFrj78cAvgX8Pqx4RkVxwd17cup8Tp47OdSl9CvOK4DRgg7tvcvcu4D5gYeYK7v60u7cFk0uAKSHWIyKSdVsb2tjd1MGpM8fmupQ+hRkEk4FtGdPbg3l9+Tjw294WmNm1ZrbczJbX1dUNYokiIuFasqkegDNmRTMI+s3MPgwsAL7Z23J3v83dF7j7gurq6uwWJyJyBJZsaqBqRDFHVY/IdSl9CjMIdgBTM6anBPPewMzeBnwRuNTdO0OsR0TyyJE8XyRbzxZxd5ZsqufNs8YNyS+SdQtzUOxlwBwzm0k6AK4APpi5gpmdBPwYuMjda0OsRUTyzJE8XyRbzxbZUt/GrsYOzpg1LivbO1yhXRG4ewK4DlgMrAEecPdVZnaLmV0arPZNYATwoJm9ZGaPhFWPiEi2PRf0D5w+xIMg1MfkuPsiYFGPeV/K+P1tYW5fRCSXnlpby+TRZRxVXZHrUg5qSHQWi4jkm454kmdfreNtx9QM6f4BUBCIiITiTxv20hFP8bb543NdyiEpCEREQvD46j2MKCnkzTOHdv8ARCgIHl25k5rLbuHZV+vY3diR63JEJI8lU84Ta2p569xqiguH/mk21M7ioSSZcmKllby8vZEVW/czp2YEFxxdQ0lRQa5LE5EQuHvO2uafebWOvS2dvPP4iTnZ/kBFJggWnjiZ99x1PV9ftIYVW/exbHMDDa1dXHriJEaWDr1Hx4nIobV1Jdjd1EFDSxcNrV3sa4vTHk/SEU/SmUhREDOKC2IUF8YYVVbEuBHFTB5dhhWXh1rXA8u2MbaimLcdM/T7ByBCQdCtuDDG6bPGMXl0GY/+ZRcPrdjB5QumUlasKwORoa6uuZNnX63jTxvqmXTNbfzk2ddeXzaipJAx5UWMKi+ltDBGSWEBSXc6E0m64in2t8dZub2RF7fuZ+qn7+ETdy7jqjNmcM6cqkG9cqhv6eSJNXu4+owZw6JZCCIYBN2mji1n4QmTeOjFHfxm5U7ee/JkCmPD46CJREVXIsXyLQ08s34vz6yvY/Wu9Hj+YyuKie/dwnmnHc+EkaVUVRZTUnjoP+YSqRR7Gju548c/4LGOc3hiTS1de7fQvOzXtLzyJKSSh/yMKVOnsW3rlj6XP/jCduJJ57IFU/tcZ6iJbBAATBpdxoXzx/PbV3bz5431nDNHA9qJ5FpLZ4Kn1tby2Cu7+MO6Olq7khTGjJOnj+Hz75jHW+dWM3/iSAoKLuSU/3P1gD67MBZj8pgy9j11O1/4x8/x6p4WXtxWQl3VZ5j9vs9xzpxqZlQd/MtfBxueorUzwU+e2cSZs8cxb0LlgGrLpUgHAcDc8ZXs3N/Oi1v3M21M+SH/EYjI4Jk6bTrbt23FCkson/cWyuedRdnMk7DCYpIt+2h7dQntm5bTsXUlG7vaeXAQt10Yi3HMxJEcPaGS1/a28uyre/n1yzuZPract86rZkx58YA/82d/3kx9axefy9JYRoMl8kEAcNbsKrbvb+eJNXu46ozp/brEFJEj4+7UdhXxru8/y7rdzXQlU4woKWR2zQhm14xg4qjZxOy0Pt8/WAPHmRmzqkcwfVwFL2/fz9JNDdyzdCunzRjLydNH97vJeHdjBz/+w0YuOLqGk6eNGZTaskVBABQWxHj7MeO5f9k2/rShnvOPrsl1SSJ5q7Etzq9f3sF9z29j4ke/x+pdTcypGcGxk0YxaXRpzm75LIgZJ08bw9zxlTyzvo7nNtWzbncz5x9dw+QxZQd9byKZ4jP3vkgi5fzzO4/JUsWDR0EQGD+ylBOnjebFrfuZN77ykAdeJF90N88cjkN1nHZzd5a+1sD9y7ax6C+76EykeNOkkdQvvpUvfu0/htT3eUaUFHLJcRN5bW8rT6+r5ZcrtnPMhEpOP2pcr7eax5Mp/uV/XuH5zQ185/IThvQDaPqiIMhwxqxxbKxt4cm1e/jgadMoLNBdRJL/whzXf1tDGw+/uIOHXtzBa3tbqSwt5LIFU7n81KkcO3kU9g/nUFL03cPadthmVlUwZcx0lr7WwEtb97O+toW540dQOv0EWjoTFJixfEsD33/yVZZt3senzjuKvzlpeD52XUGQoaggxvlH1/DwSztZtnkfZxw19McIkaHjSP6yhv7/dT2UuTvr9jTzh3V1PLmmluc3NwDw5plj+fT5s7n42InD6js7RQUxzppdxfFTRvHC5n2s2d3E+Cu+xrFfXvz6OqPLi/j2ZSfw3pOHZwiAguAA08dVcPSESpZvaWDu+BGMG1GS65JkmDiSv6whe0/NOlzJlNOVTNGV+OurbNYCfvnCdl6tbWb97mZW7Wyitjn9xNl54yu58cK5vOekyUwZE+43ecM2srSI846u4czZVdz8qQ/xrZ/eT0HMmDa2nAuOqRn2N5goCHpx9pwqNte38uTaWj5wypQhP5a4yGBp7Uywq7GD3Y0dNLbHae6M09KZoCOeIpnyA9av+cDN3PjgyxQXxJhVXcEZR43jzKOqOHtuFRNH5V8/W3FhjI7XVvCp82bnupRBpSDoRXlxIWfPrubxNXt4ZUcTx00ZleuSRELh7hRVz+RPG/ayobaF/e1xIH0HzcjSQipLixhXUUJZUQHFhekxe7rH7ikujPHD6y9n7coVTB1Tpj61YUxB0IdjJlayZncTf9y4l1nVFVSU6D+V5I/WzgS/WrGdu5/bwqSP/ScvbN3HtDHlHDdlFBNHlVJTWUpB7NBXwl271jNTX8Ic9nR264OZcf7RNdyzZCu/X1/HO48bHsPJihzMjv3t3Pnnzdz7/FaaOxKcMGUU9Ytv5aZ//RblxTodRJWO/EGMKS/mtJljeW5TPev3NOe6HJHD4u6s2LqPO/64mcdW7Qbg4mMn8LGzZnLytDHYp39LefHQvIVTskNBcAgLpo9h094Wnl5bS0HF8PrauIQvlXK2NLSxeW8rFfPPZfXOJooLY1SWFjKmvDinwxC3dSX49Us7ufu5Laze1cSosiKuOXsWV58xnUmj868jVw6fguAQYjHjwvkTuPf5rVS9+0aSKe9X26nkr9bOBI+v3sNvX9nFkk0NNAYdrFXvvpHH1+x5fT0zmDCylKljypk2tpyJo0uJZeEOtPV7mvnF0q386oXtNHcmOHpCJf/6nmN578mT1fwjvdK/in4YW1HMufOqeSJ1At97Yv2Qv99bwlHX3Mmdf97M3Uu20NgeZ8LIUi560wROmT6Go2oqOP2k4/jiz35HZzxJY0ecuuZOtjW0s2xzA89vbqCsqICjaiqYU1PJlNFlxAbpDwp357W9rSz6yy5+8/Iu1u1pprggxiXHTeCqM6anm390C7QchIKgn940aRQPP3AP3wfmTqjkXcdPynVJkiUb61q4/dlN/GrFDuLJFO+YP4G/PXMGp84Y+4aTeWLfTkaVFUFZETUjS5lTUwlHQWciydb6NjbUtrBudzOv7GiitCjGUdUjmF09gkmjywbchLSnqYMXt+7jjxv28of1dWxraAegY9sq2tY+S+vaZ/leWyPfG9T/EpKvFAQDUL/4h5z7rsu44YGXGT+ylFNnjM11SRKiF7Y08OM/bOLxNXsoKojx/lOmcM3ZswZ8u2RJYQFzxlcyZ3wliWSKzUEovLqnhVU7mzCgqrKEsW//e37yzCZqRpZQXZn+Rns86XTGk+xq7GBbQxub69tYtbORXY0dAJQXF/CWo8bx0n3/wee/8u9Uls4B3jOg+nSFO0AWy7srLAXBQCTj3Hb1At7/oz/z0Tue586PncYChUFe6UqkWLxqN3f+eTPLt+xjVFkR1503m6vPmPH6yflIFBbEXh9vP5FMsWN/Ozv3d7CzsZ2KN53P1xat6fO9pUUxpo4p59QZYzlx6mhOmDqK4yaPprgwxk8/uojK0u8ccX0DlocnxUPyVGiD9OWKgmCAxlYUc++1p3PlbUu46qfP853LT+CiY/Udg95kY3jjweDurN3dzKMrd3L/sm3sbeli6tgybn73fC47dWpoHayFBTGmj6tg+rj0FcYNF15AY3sXexo7qGvuxMwoLoxRUhhLXyWMKBl6J90jOCnC0D0xRo2C4DCMH1nKfX93Otfe9QJ///MV/N05s7j+7XMpHUJjqg8FmYOwuTsdiRTtXUnau5K0xRO0dyXpSqTAwDAMwNIjPt7/jRt49tU6RpYWMaqsiJFlRYwsLRyUYQxSKWdzfSurdjaxfHMDT6ypZcf+dmIG5x89ng+fPo1z5lQPWmfuQIwsLWJkaRFzxg+f593K8KcgOEw1laXcd+3p3PLoan78zCYeX72HG98xj4uPnTD0/mobIHcnmXISqZ4/0wOPJZLp6XgyRVNHgqaOOE3tcZo6Euxr7aKhtYv61i5qLv8aP1+6JX3yjyfxA8cs61P1wpu46qfPHzC/orggCIXugChkZGkRlaWFFBfGKCyIcesP/pOm/fsAsMJirLCEWEkZBSPGUTBiLIUja4iVpEfDTMU76Nj8Eu0bltK2cRmLx47k9mE+FLTIQCkIjkBpUQH/9jfH8Y43TeCW36zik/esYGZVBe8/ZQoXHFPDvPGVgxYK7k5nIkVbV5LWzgStXQlaO5O0BT9bOxPp37uXB8taOhO0dSXpiCeJJ1N0JZ2uRIp4MnglUq8PLRxPpk/uiV5GmRyIVEcLybZGrLCI0WVFTBxZSllxAWVFBZQVF1BeXEhZUQHlxQWv3y3jDo6DpztIv/qxS/jTshU0tcdpDF5N7Zmhk563c38HazuaaWqPEw8Cyua/g9Gx9NVZgRkFsXQTS0VJARXFhVSWFlJdWUJNZSljK4opiB0HXAXkuKkiiu3tMiSEGgRmdhHwPaAAuN3dv95jeQlwF3AKUA9c7u6bw6wpDG+dW83vrn8rj67cyT1LtvLNxev45uJ1lBcXpG8RrBlBdWUJ5cXpk1/MjM5E6vUTclciRWtX90k94wT/+kk9fTLv7wnaDCqKCykvLmDPjq10tTXjiU48kYBUHE8mglccMn73ZBxSSTyZSP9MJXn3J24kZhAzw4Kf6Vf6y3bdbdilhemTeklR7PWHfd9w4Tz+6SMDbz8uKYL43i2HfVeWmfEfi9fikJUvcA2aPOyElOEhtCAwswLgVuDtwHZgmZk94u6rM1b7OLDP3Web2RXAN4DLw6opTAUxY+GJk1l44mR2Nbbz7Pq9rN7VxMa6FpZuqmdHfRNWWHzA+14/CXd1kIq3413tpLo68Hjws6sdT3SS6mwj1RUsj3cE67X3+j6Pd75hG0dycjnlq187rPfmmlnQ5yAihxTmFcFpwAZ33wRgZvcBC4HMIFgI3Bz8/kvgB2Zm7gNpTR56Jo4q47JTp75hnpnxrcVrSSQdx19vsuhPU8ANF86L3l+KuWomUfOMRJCFdc41s/cDF7n7J4Lpq4A3u/t1Geu8EqyzPZjeGKyzt8dnXQtcG0zOAw73frUqYAmcKy4AAAc+SURBVO8h18ov2udo0D5Hw5Hs83R3r+5twbDoLHb324DbjvRzzGy5uy8YhJKGDe1zNGifoyGsfQ5zjNwdQGb7yJRgXq/rmFkhMIp0p7GIiGRJmEGwDJhjZjPNrBi4AnikxzqPAB8Jfn8/8NRw7x8QERluQmsacveEmV0HLCZ9++gd7r7KzG4Blrv7I8BPgbvNbAPQQDoswnTEzUvDkPY5GrTP0RDKPofWWSwiIsND7p6jJyIiQ4KCQEQk4iITBGZ2kZmtM7MNZnZTrusJg5lNNbOnzWy1ma0ys38I5o81s8fN7NXg55hc1zqYzKzAzF40s0eD6ZlmtjQ41vcHNyvkDTMbbWa/NLO1ZrbGzM6IwDG+Pvg3/YqZ3Wtmpfl2nM3sDjOrDb5f1T2v1+Nqad8P9n2lmZ18JNuORBBkDHdxMTAfuNLM5ue2qlAkgM+5+3zgdOBTwX7eBDzp7nOAJ4PpfPIPQOYTXb4BfMfdZwP7SA9lkk++Bzzm7kcDJ5De97w9xmY2GfgMsMDdjyV980n3kDT5dJx/BlzUY15fx/ViYE7wuhb40ZFsOBJBQMZwF+7eBXQPd5FX3H2Xu68Ifm8mfYKYTHpf7wxWu5OBPstwCDOzKcA7gduDaQPOJz1kCeTf/o4CziF9xx3u3uXu+8njYxwoBMqC7xuVA7vIs+Ps7s+QvnsyU1/HdSFwl6ctAUab2WE/ISsqQTAZ2JYxvT2Yl7fMbAZwErAUGO/uu4JFu4HxOSorDN8F/hFIBdPjgP3ungim8+1YzwTqgP8OmsNuN7MK8vgYu/sO4FvAVtIB0Ai8QH4f5259HddBPadFJQgixcxGAL8CPuvuTZnLgi/s5cU9w2b2LqDW3V/IdS1ZVAicDPzI3U8CWunRDJRPxxggaBdfSDoEJwEVHNiEkvfCPK5RCYL+DHeRF8ysiHQI3OPuDwWz93RfNgY/a3NV3yA7E7jUzDaTbu47n3T7+eigCQHy71hvB7a7+9Jg+pekgyFfjzHA24DX3L3O3ePAQ6SPfT4f5259HddBPadFJQj6M9zFsBe0j/8UWOPu385YlDmUx0eAX2e7tjC4+xfcfYq7zyB9TJ9y9w8BT5MesgTyaH8B3H03sM3MuscXv4D00O55eYwDW4HTzaw8+Dfevc95e5wz9HVcHwGuDu4eOh1ozGhCGjh3j8QLuARYD2wEvpjrekLax7NIXzquBF4KXpeQbjd/EngVeAIYm+taQ9j3c4FHg99nAc8DG4AHgZJc1zfI+3oisDw4zg8DY/L9GANfAdYCrwB3AyX5dpyBe0n3gcRJX/l9vK/jChjpOyE3An8hfUfVYW9bQ0yIiERcVJqGRESkDwoCEZGIUxCIiEScgkBEJOIUBCIiEacgkGHPzNzMfp4xXWhmdd2jkR7kfSea2SUZ0zeb2Y0D2O6MzJEiD7LOBzOmF5jZ9/u7jYN8bomZPRaMxvnJjPm3HelIlBI9CgLJB63AsWZWFky/nf59y/JE0t+zCNMM4PUgcPfl7v6ZQfjcdwB/BI4HrgIwsxOAAg8GHhTpLwWB5ItFpEchBbiS9JdzADCzimCs9+eDgdoWBt8wvwW43MxeMrPLg9Xnm9nvzWyTmX0meP8tZvbZjM/7mgXPesiYN8PMnjWzFcHrLcGirwNnB9u43szOtb8+N2GsmT0cjCe/xMyOD+bfHNT7hjp6iJMehbOI9JeLAL4K/N/D/Q8o0aUgkHxxH3CFmZWS/it5acayL5IefuI04Dzgm6RPoF8C7nf3E939/mDdo0n/tX0a8OVg7KY7gKsBzCxGejiLn/NGtcDb3f1k4HKgu/nnJuDZYBvf6fGerwAvuvvxwD8Dd2Us662OTI+TvtpYAnzfzC4FVrj7zoP/ZxI5UOGhVxEZ+tx9ZTD09pWkrw4yXUh6cLru9v9SYFofH/W/7t4JdJpZLelhgDebWb2ZnUR6GOAX3b3ezCoz3lcE/MDMTgSSwNx+lH0W8L6g/qfMbJyZjeyrDtLDDnTvb4KgySkIicXAQjP7drBvd7l73o2nJeFQEEg+eYT0uPXnkh6jpZsB73P3dZkrm9mbe/mMzozfk/z1/5HbgY8CE0hfIfR0PbCH9BPDYkDHgKvvXx29+STpq4nTSY/VfznwFHk4sKKEQ01Dkk/uAL7i7n/pMX8x8Olg5EqCv+wBmoFK+ud/SI+Bf2rweT2NAna5e4p0521BP7bxLPChoKZzgb3e4/kRhxKM1f8u0kFQTvoBPQ6UHex9IpkUBJI33H27u/d2a+ZXSTfdrDSzVcE0pIcxnt+js7ivz+4K1n/A3ZO9rPJD4CNm9jLp9v3WYP5KIGlmL5vZ9T3eczNwipmtJN2p/BEG7kvA14IAWgycTXo0yrsP47MkojT6qEg/BJ3EK4APuPurua5HZDDpikDkEMxsPukx759UCEg+0hWBiEjE6YpARCTiFAQiIhGnIBARiTgFgYhIxCkIREQi7v8DDL8VeBRLb/oAAAAASUVORK5CYII=\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sns.histplot(data=SRR5836475, x=\"Methylation %\", bins=20, kde=True, kde_kws={'bw_adjust': 2})\n",
        "plt.title(\"ICM\")\n",
        "plt.savefig('ICM.png')"
      ],
      "metadata": {
        "id": "NKdnAGyad5Ov",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "outputId": "c60846de-fee5-4e57-cbc6-c3087bee6dd4"
      },
      "execution_count": 65,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxddZ3/8dfnLtm3tknXpAu0LGVpC2UXRVFsGaEjKosLqCzjCIKC4zjjDCKOs6g/VEbUYRBZVBCRpQ4oKiCLSO1e6QahLW26JU2bJk2z5/P7456US0iaNM3NTe55Px+PPHrPcs/5nJz0vu9Zvt9j7o6IiIRXJN0FiIhIeikIRERCTkEgIhJyCgIRkZBTEIiIhJyCQEQk5BQEIiIhpyAQ6Qcz22Rm7w1eTzCzH5vZdjNrMLN1ZvY1M8sPpruZVZtZLOn98WCcGu7IsKMgEDkEZjYa+DOQC5zh7oXA+4AS4MikWfcA85OG5wfjRIYdBYHIobkRaAA+7u6bANx9i7vf4O6rkua7H7g8afhy4L4hq1LkECgIRA7Ne4FH3L2zj/keA95pZiVmNgo4G3g85dWJDMCIDAIzuzs43/pKP+e/2MzWmNlqM/t5quuTjDYG2N6P+ZqBXwOXBD8Lg3Eiw06s71mGpXuA79OPQ20zmwH8E3CWu+8xs7Eprk0yWy0woZ/z3gf8B2DAP6asIpHDNCKPCNz9eWB38jgzO9LMfmtmS83sBTM7Jph0NXCHu+8J3ls9xOVKZvkD8EEz68//nRdIhMY44MWUViVyGEZkEPTiTuBz7n4y8EXgB8H4o4CjzOxPZvaymc1LW4WSCW4DioB7zWwKgJlNMrPbzOzE5Bk90cf7BcCFrv7eZRgbqaeG3sLMCoAzgV+aWdfo7ODfGDADOAcoB543sxPcvW6o65SRz913m9mZwL8Bi4K2A1uBB4DKHuZfPcQlihyyjAgCEkc2de4+u4dpVcAid28DNprZqySCYfFQFigjm7tPTXq9Dfj0Qea1XsZXkrheIDKsZMSpIXevJ/Eh/xEAS5gVTH6MxNEAZlZK4lTRhnTUKSIyHI3IIDCzB0i07jzazKrM7ErgY8CVZrYSWA0sCGZ/Cqg1szXAs8A/uHttOuoWERmOTNewRETCbUQeEYiIyOAZcReLS0tLferUqekuQ0RkRFm6dOkudy/radqIC4KpU6eyZMmSdJchIjKimNkbvU3TqSERkZBTEIiIhJyCQEQk5BQEIiIhpyAQEQk5BYGISMgpCEREQk5BICIScgoCEZGQC1UQVEyegpkN+Kdi8pR0b4KIyKAbcV1MHI6qLZu57XfrB/z+G887ehCrEREZHkJ1RCAiIm+nIBARCTkFgYhIyCkIRERCTkEgIhJyCgIRkZBTEIiIhJyCQEQk5FIWBGZ2t5lVm9krvUz/mJmtMrO/mtlLZjYrVbWIiEjvUnlEcA8w7yDTNwLvcvcTgK8Dd6awFhER6UXKuphw9+fNbOpBpr+UNPgyUJ6qWkREpHfD5RrBlcBveptoZteY2RIzW1JTUzOEZYmIZL60B4GZvZtEEPxjb/O4+53uPtfd55aVlQ1dcSIiIZDW3kfN7ETgLmC+u9emsxYRkbBK2xGBmU0GHgE+4e6vpqsOEZGwS9kRgZk9AJwDlJpZFfBVIA7g7j8CbgbGAD8wM4B2d5+bqnpERKRnqbxr6LI+pl8FXJWq9YuISP+k/WKxiIikl4JARCTkFAQiIiGnIBARCTkFgYhIyCkIRERCTkEgIhJyCgIRkZBTEIiIhJyCQEQk5BQEIiIhpyAQEQk5BYGISMgpCEREQk5BICIScgoCEZGQUxCIiIScgkBEJOQUBCIiIacgEBEJOQWBiEjIKQhEREJOQSAiEnIpCwIzu9vMqs3slV6mm5ndbmaVZrbKzE5KVS0iItK7VB4R3APMO8j0+cCM4Oca4IcprEVERHqRsiBw9+eB3QeZZQFwnye8DJSY2YRU1SMiIj1L5zWCScCWpOGqYNzbmNk1ZrbEzJbU1NQMSXEiImExIi4Wu/ud7j7X3eeWlZWluxwRkYySziDYClQkDZcH40REZAilMwgWApcHdw+dDux19+1prEdEJJRiqVqwmT0AnAOUmlkV8FUgDuDuPwKeBM4HKoH9wKdSVYuIiPQuZUHg7pf1Md2Ba1O1fhER6Z8RcbFYRERSR0EgIhJyCgIRkZBTEIiIhJyCQEQk5BQEIiIhpyAQEQk5BYGISMgpCEREQk5BICIScgoCEZGQUxCIiIScgkBEJOQUBCIiIacgEBEJOQWBiEjIKQhEREJOQSAiEnIKAhGRkFMQiIiEnIJARCTkFAQiIiGnIBARCbmUBoGZzTOz9WZWaWZf7mH6ZDN71syWm9kqMzs/lfWIiMjbpSwIzCwK3AHMB2YCl5nZzG6z/QvwkLvPAS4FfpCqegAi2fmpXLyIyIiUyiOCU4FKd9/g7q3Ag8CCbvM4UBS8Lga2paqY376yg4mf+TE79janahUiIiNSKoNgErAlabgqGJfsFuDjZlYFPAl8LlXFzJlcQmdTPY+t2EpNQ0uqViMiMuKk+2LxZcA97l4OnA/cb2Zvq8nMrjGzJWa2pKamZkArGleUw84Hv0I8GuHZ9dWHV7WISAZJZRBsBSqShsuDccmuBB4CcPc/AzlAafcFufud7j7X3eeWlZUNuKCO+hpOmFTM9r3N7GtpH/ByREQySSqDYDEww8ymmVkWiYvBC7vNsxk4F8DMjiURBAP7yt9P08cWAPB69b5UrkZEZMRIWRC4eztwHfAUsJbE3UGrzexWM7swmO0m4GozWwk8AHzS3T1VNQGMzs9idH4WlQoCEREAYqlcuLs/SeIicPK4m5NerwHOSmUNPZleVsDiTbvZ39pOXlZKfwUiIsNeui8Wp8W00nwc2FrXlO5SRETSLpRBUFqYRcRgZ71uIxURCWUQxCIRSguyqa5X4zIRkX4FgZm97Tx+T+NGkrFF2exsaCHF16ZFRIa9/h4R/Hc/x40Y44pyaG3vpK6pLd2liIik1UFvmTGzM4AzgTIzuzFpUhEQTWVhqTauMAeAnfXNjMrLSnM1IiLp09cRQRZQQCIwCpN+6oEPp7a01BqTn0UsYrpgLCKhd9AjAnd/DnjOzO5x9zeGqKYhEYkYZYXZVDfogrGIhFt/W1Nlm9mdwNTk97j7e1JR1FAZnZ/FhprGdJchIpJW/Q2CXwI/Au4COlJXztAanZ/F6m31NLV1kBsf0Zc8REQGrL9B0O7uP0xpJWnQdZF4T2MruSW5aa5GRCQ9+nv76K/N7LNmNsHMRnf9pLSyITA6PxEEu/e3prkSEZH06e8RwRXBv/+QNM6BIwa3nKFVmBMjGjH2NCoIRCS8+hUE7j4t1YWkQ8SMUXlxdisIRCTE+hUEZnZ5T+Pd/b7BLWfojcrLolrPMBaREOvvqaFTkl7nkHiq2DJgxAdB10Nq2js6iUVD2QefiIRcf08NfS552MxKgAdTUtEQG5WXhQN1TW2UFmSnuxwRkSE30K/AjUBGXDcYlR8H0AVjEQmt/l4j+DWJu4Qg0dncscBDqSpqKBXnJoJgb7N6IRWRcOrvNYJvJ71uB95w96oU1DPksmNRcuNR9u5XEIhIOPXr1FDQ+dw6Ej2PjgIy6jxKUW6MvXougYiEVH+fUHYx8BfgI8DFwCIzG9HdUCcrzo0rCEQktPp7augrwCnuXg1gZmXAH4CHU1XYUCrJzeK16n10dDrRiKW7HBGRIdXfu4YiXSEQqD2E9w57xblx3KFBF4xFJIT6+2H+WzN7ysw+aWafBJ4AnuzrTWY2z8zWm1mlmX25l3kuNrM1ZrbazH7e/9IHz4E7h3R6SERCqK9nFk8Hxrn7P5jZRcA7gkl/Bn7Wx3ujwB3A+4AqYLGZLXT3NUnzzAD+CTjL3feY2diBb8rAKQhEJMz6OiL4LonnE+Puj7j7je5+I/BoMO1gTgUq3X2Du7eSaIm8oNs8VwN3uPueYB3VpEF+dpRoxBQEIhJKfQXBOHf/a/eRwbipfbx3ErAlabgqGJfsKOAoM/uTmb1sZvN6WpCZXWNmS8xsSU1NTR+rPXRmRnGO7hwSkXDqKwhKDjJtMB7pFQNmAOcAlwH/G/Rj9Bbufqe7z3X3uWVlZYOw2rcrzlMQiEg49RUES8zs6u4jzewqYGkf790KVCQNlwfjklUBC929zd03Aq+SCIYhV5QTo76pHXfve2YRkQzSVzuCzwOPmtnHePODfy6QBXywj/cuBmaY2TQSAXAp8NFu8zxG4kjgJ2ZWSuJU0Yb+lz94inLjtHZ00tLeSY4eZC8iIXLQIHD3ncCZZvZu4Phg9BPu/kxfC3b3djO7DniKREd1d7v7ajO7FVji7guDaeeZ2RqgA/gHd689jO0ZsKKcxJ1D9U1tCgIRCZX+Po/gWeDZQ124uz9Jt/YG7n5z0msHbgx+0qooJ/GrqG9uZ2xRmosRERlCGdM6+HAVBW0J6tW6WERCRkEQyI5FyIpGqNedQyISMgqCgJlRlBujvrk93aWIiAwpBUGSopy4Tg2JSOgoCJIU5cSpb2pTWwIRCRUFQZLC3BhtHU5ze2e6SxERGTIKgiRdvZDqgrGIhImCIMmBRmW6TiAiIaIgSNLVqKyhSXcOiUh4KAiSZMejZMUiOiIQkVBREHRTlBNTd9QiEioKgm6KcuI0qFGZiISIgqCbotxEozK1JRCRsFAQdFOUE7QlaFNbAhEJBwVBN+qFVETCRkHQTfIDakREwkBB0E3yA2pERMJAQdBNdjxKdkzPJRCR8FAQ9KDrziERkTBQEPSgKEcPqBGR8FAQ9EDPJRCRMFEQ9KAoN057p9PU1pHuUkREUk5B0IMDdw6pF1IRCYGUBoGZzTOz9WZWaWZfPsh8HzIzN7O5qaynv9SoTETCJGVBYGZR4A5gPjATuMzMZvYwXyFwA7AoVbUcqsIDbQkUBCKS+VJ5RHAqUOnuG9y9FXgQWNDDfF8H/gtoTmEthyQ7FiUnFtGpIREJhVQGwSRgS9JwVTDuADM7Cahw9ycOtiAzu8bMlpjZkpqamsGvtAdqSyAiYZG2i8VmFgFuA27qa153v9Pd57r73LKystQXR+L0kFoXi0gYpDIItgIVScPlwbguhcDxwB/NbBNwOrBwOF0wrm9uV1sCEcl4qQyCxcAMM5tmZlnApcDCronuvtfdS919qrtPBV4GLnT3JSmsqd+Kc+J0dDr7W9WWQEQyW8qCwN3bgeuAp4C1wEPuvtrMbjWzC1O13sFSmJu4c0iPrRSRTBdL5cLd/UngyW7jbu5l3nNSWcuh6nouwd6mNsYX56S5GhGR1FHL4l4cCALdOSQiGU5B0IusWIS8rCh79ysIRCSzKQgOojg3zl7dQioiGU5BcBAlCgIRCQEFwUEU58bZ19JOe0dnuksREUkZBcFBFOe9eeeQiEimUhAcRHGugkBEMp+C4CBKcrMABYGIZDYFwUHkxCNkRSMKAhHJaAqCgzAzivPi1CkIRCSDKQj6UJwbV6MyEcloCoI+lAQPqOnoVHfUIpKZFAR9GJWfRaejh9SISMZSEPRhdF7izqE9+1vTXImISGooCPowKmhUtltBICIZSkHQh+x4lLysKHsadWpIRDKTgqAfRuVl6dSQiGQsBUE/jMqLKwhEJGMpCPphVH4WzW2dRHKL0l2KiMigUxD0w6jgzqH46ElprkREZPApCPphdH4QBGPK01yJiMjgUxD0Q2FOjFjEiJdOSXcpIiKDTkHQDxEzRudnES+bmu5SREQGXUqDwMzmmdl6M6s0sy/3MP1GM1tjZqvM7GkzG7ZfucsKs8kaOw139TkkIpklZUFgZlHgDmA+MBO4zMxmdpttOTDX3U8EHga+map6DldpQTbRvGJq9rWkuxQRkUGVyiOCU4FKd9/g7q3Ag8CC5Bnc/Vl33x8MvgwM26uxpQWJC8brtjekuRIRkcGVyiCYBGxJGq4KxvXmSuA3PU0ws2vMbImZLampqRnEEvtvTEE2AOt21Kdl/SIiqTIsLhab2ceBucC3epru7ne6+1x3n1tWVja0xQVy41HaG3bpiEBEMk4qg2ArUJE0XB6Mewszey/wFeBCdx/WJ+DbqjexdoeCQEQySyqDYDEww8ymmVkWcCmwMHkGM5sD/A+JEKhOYS2DorVmI5XVDbS0d6S7FBGRQZOyIHD3duA64ClgLfCQu682s1vN7MJgtm8BBcAvzWyFmS3sZXHDQuv212jrcNbq9JCIZJBYKhfu7k8CT3Ybd3PS6/emcv2DrWXbegCWb97D7IqSNFcjw1HF5ClUbdk8oPeWV0xmy+Y3Brkikb6lNAgyTce+WsYX5bBiS126S5FhqmrLZm773foBvffG844e5GpE+mdY3DU0ksyZXMLyzQoCEckcCoJDNLuihM2791OrFsYikiEUBIdozuRRADo9JCIZQ0FwiE6YVEw0Yix9Y0+6SxERGRQKgkOUmxVldkUJf6rcle5SREQGhYJgAN4xvZRVW/dSpwfaH1TF5CmY2YB+KiYP2x7JRTKObh8dgHceVcr3nn6Nl16v5fwTJqS7nGFLt1KKvN1wbGuiIBiAWeUlFGbHeOG1GgVByLV1dLJow24WbaylvqmNkndezqbaRspH5RKL6IBb3m44fkFSEAxALBrh9CPH8MJru3B3zCzdJckQc3ceWbaV7z79Klt2NxGNGIU5MYpO/SCPr9hGQXaMU6eN5viJRfr7kGFPQTBA7zqqjN+v2cm6HQ0cO6Eo3eXIENq1r4UvPbyKZ9ZVc8KkYv75Y8fyzqPKyM+OEYnncO29L7HkjT08s66a16obeP/M8eRn67+aDF86dh2g+cePJxoxFq7clu5SRqSW9g527Wtha10TNQ0ttLZ3prukfnl1ZwN/e8efeLFyF1+9YCaPX3sW80+YcOCD3ttbOKKsgI+cXM65x4xle10zDy7ewi41QJRhTF9TBmhMQTZnTS/l1yu38aX3H52xh/+Hc2Gru5qGFtbvbGDTrkZqG996x5UBYwqymDG2kGMnFA7K+gbb0jd288mfLCYnHuXhz5zBieW9dzxoZhw/qZhxRTk8vmIrv1xaxYJZE5lYkjuEFYv0j4LgMCyYNZGbfrmSZZvrOHnKqHSXkxKHe2GrvaOT16r3sapqLzvqm4kYlI/KY8a4AkbnZZEVi9Da3kltYytb9uznzxtqWbSxljHzr2f73iYmFA+PD86XKndx1X1LGFeUw0+vOo1J/fxALyvM5uJTKnh02VYeW7GVBbMn9fu9Il1a2jrY19JOtHBMSpavIDgM5x03juxHIzy8tGrEBkFnp7OrsYUde5upbWyls9Pp6HSiEaOsMJtowRg6O51I5NCOeKobmhl17jX8+MWNNLd3MiovzruOKuOY8YXkxKNvm38GcDpj2NvUxvLNe1jRdg7v+fZzXPvuI7nq7CN6fM9QeXZdNZ/56VKmjsnn/qtOZWxhziG9vygnzodOLueRZVU8vmIrF86aSPmovBRVK5mivqmNlVV1bNjVSN3+NgAK53wgJetSEByGwpw4F500iV8tq+LG9x1FWWF2uks6qOr6ZlZvr2fd9gbWbq9n3Y56Nu5qpK3De31P+bX38v0/VlKUE6ckL86ovCxKchOvS/KyyIpG6HSnua2DuqY2tu9tZlNtI7X7WimcPZ+K0XkcP6mYilG5/Tp9Vpwb55yjx/Lkv3yYK//7Cb79u1f5xZIt/MvfzOS8meOG/BTcwpXbuOmhFRwzvoj7Pn0qo/KzBrScguwYHzqpnEeWbeXxFdu4cNZEKkYrDOTtmlo7eOn1XazeXo8BFaPzOG5iEYXZcX5819MpWaeC4DBd884jeXDxFn7yp418ad4x6S7ngMaWdlZV7WVlVR0rNtexYksdO+qbD0yfVJLLsRMKec8x45hUksP44lzGFGQRixgRM9o7nZqGFi766CeZ95l/pW5/G3VNbWyr23vQ4IgYjC/O4Zyjyrj/2nP5/MKlA6q/fe9OfvSJk3mpche3/Ho1f3f/Us6eUcpXLziO6WMLBrTMQ3X3ixu59f/WcNq00fzvFXMpyokf1vLys2NcdNIkHl2+lcdXJsJgssJAkrxes48/rNlJa0cns8pLOGlyCYVJf3fttVUpWa+C4DBNK81n/vHjuf/lN7j67CMG/I3xcG3f28SLr+1i6Rt7WLGljld3NtAZfF5PGZPHaUeMZlZ5CcdPKubo8YUU5/bvQ23fyt9y5pHfOzDs7uxv7WDP/lb2NrXRHoRCdjxCSW4WYwqyiEcTN6Pd27zvsLfrzOmlPHn92dz/8hvc9vtXmffd5/nUWVO5/twZb/kPMpg6O51vPrWeHz33OvOOG893L509aKemusLgkeVbWbhyGxecOIEpY/IHZdkycrV3dPJi5S5WVu1lbGE2580cx5iCoTvDoCAYBDecexS/W72Trz+xhtsunj0k62xobuPlDbt58bUaXqzcxes1jUDi1MqsihLOO248cypKuPyCc3i+ci3PD9J6zYz87Bj52THKh+iySCwa4VNnTeOCWRP59lPruevFjfxyaRWXnzGVK86YMqj/YfY0tvKFh1bwx/U1fOy0ydy64Hiih3h9pC95WTE+NKecR5dv5dcrt/P+48YxY9zwvFNKUm/P/lZ+89cd1OxrYU5FCWdOHzPkrdIVBIPg6PGFfPacI7n9mUouOHEi7z5m7KCvo62jk+Wb63ixchd/qtzFii11dHQ6OfEIp00bw6WnTOYdM0o5elzhWy7sVlWuHfBdPzC8+vwpLcjmPz90Ih89bTK3P13J7U+/xp3Pv84H55SzYPZETpk6esAf2p2dzsKV2/i3J9ZQ39TO1//2eD5+2uSUXZPIzYpy0UmTWLhyG0++soPTGlvB0tOsZzj2fRMWa7fX8+z6aqIR44JZEziidGhOe3anIBgk175nOr9bs5PrH1jOz68+nRPKiw9ree7Oa9X7ePG1XbxYuYtFG2ppbO0gYnBCeQmfedcRvGN6GSdNKSE7lr47atLhxPIS7rpiLpXVDdz5/AYeXV7FA3/ZTFlwSH3qtNGcPGUUk0r6vkDd0NzGU6t3ctcLG1i3o4FZFSXc9+kTmDkx9a3Fc+JRLpoziafXVbNo427GXfoNtuzeP+QXkbfV1nPDA8uobWylpa2TDnc6O52sWIT87BgF2bHEzQG58bf9PofTF4WRpKW9g+derWHt9gYmleTy/uPGpexUZ38oCAZJdizKTz51Ch/+4Z/5xN2L+M7Fsw/pyMDdeaN2Py9vqOXlDbW89Hot1Q2J1qhTx+TxwZMm8Y7ppZxxRCnFeen7gxlOpo8t5JsfnsUtFx7HM+uqeWLVdh5bvpWfLUp8u+3Yt4e2Pdtor9tOe301nS37wTuxWDaxwlLiY6eRPX4GFovTVruFvX9+iGUNG5h57aYh24ZYNMJ5M8dRMSqPp1r2c+5tz3HlO6bx6bOmDfpdaK3tnbxes491O+pZG9w5tnZ7AxWf+ymPrXizhXw0YkSMt90UkBWNMLYom3FFOUwszmF88aHdRisJuUeeyk9f3kxjSzunTRvNqdNGE0lzg1QFwSCaUJzLz68+jWvuW8qn7lnM/OPH8/HTp3DylFFvudjo7uzZ38a67fWs3lbPK9v28peNu9m+N3FXT2lBNqcfMZqzZ5Ry5pGlnDn7WJ4bpNa9mSgvK8YHTpzIB06cSHtHJ3kTZ/DR/3iA6oYi9jZNZG9TG/ta2t/ynuxYhNH5WUwozmH62ALGF03HLn5PWr7hmhkzJxbxk+v+nuv+9/f86LnX+fGLG3nfzHHMP348p04bfUhtF5rbOqjas5/XaxrZuKuRV3c0sHZHA5XVDQc+3LNiEY4aV8A5R5dx5zdv5qovfo3SwmxyYpED3/o7Op39re3sa2mntrGV6voWqhuaWb55D0uDjJh49f9w00MrOXnKKE4sL2Zaab76VerFtrom/v3JtYz98M1kxyP8zYkVjC8aHmGqPTbIpozJZ+HnzuL7z1Ry70ub+M0rO4hFjLb6GtqaGonkFBDNK8KiSbeENeyipWoNzZv/SvPmv/LG7iqWAnckLXe4dVs7XMWiEdqqNzCr4q3dP3S609bRiTvEIkYsOvy62erYV8t3L53D586dwb0vbeKJVdt5YtV2AEry4kwszmViSS4F2VGikQjRSOJb/r6WDva1tLGnsY2dDc0HGh91GVuYzbETinjXUWUcO6GQYycUcURp/oHfwf+7eCEVo7/1tnoSParGKcyJJ1p4T0yMb+/oZGdDC9v3NvH7J//MH9dP41fL3rytcUJxDhWj8ygtyGJMfjYleXFy4lGyY5HgJ0p2PPE6LytGfnaUvKwYeVnRA8O58WjGdNuycVcjd7+4kV8s3gJA3Qs/5bqbbxn0mxAOR0qDwMzmAd8DosBd7v6f3aZnA/cBJwO1wCXuvimVNQ2F7FiUm847mmvfPZ3nXq1h5ZY6vvn9p5n17gvIjkWDP/goo/OzKCvMJi9rBnBGr8sL24d5KkTMRsy1lCPLCrh1wfHc/IGZrNq6l6Wb9vDG7ka21TVTtWc/+1s76AhagHedx1+zchlNdTV07Kulfd9uOupraNu9lbbdW3mjpZHFg1hfLBphUkkuk0py+fmj32Dnr77Optr9rN1ez4aafWyoaaSqrolXd+6jdl8tdU1teO9NT3rkHW10NjfS2byPzpZ9b75u3kdnS+J1cV4W/3P7bRTlxikOfopy4xRmxw65Jfxgau/oZO32BhZtrOWp1TtYvGkP8ahx0Zxyrn/vDMr//Xyika+lrb6epCwIzCxK4kvt+4AqYLGZLXT3NUmzXQnscffpZnYp8F/AJamqaajlxKO8/7jxvP+48fzj/Nv4wA1/l+6SZASJRSOcNHkUJ03u+z5ds3em7ajRzJhWms+00p7bQ7g7bR1OS3sHLe2diZ+2DpraOjj51DP4zLd/RltH4oitraOT1q55gvma2zsT721LjGtu7zgQLH//s2U9rK+Tzpb9bwZHcyOdLfvw1ia8vQ1vb6EgL4ebbrienHiEnHiUnHjiSKXr30jEiFriWokF/0aCayedDvtbO2hq7aCprZ39rR3srG+has9+qvY0sXrrXhpbOwCYMbaAL3DVeYcAAAf2SURBVJ53FBefUnHIXZMMpVQeEZwKVLr7BgAzexBYACQHwQLgluD1w8D3zczcD/X7wxCxSMYcrg576fpdp3Mfj8S/r0Go+VAb1HUFy1cuOZsv3fPMgYBpbguCpi0RHM1BkHSNa+vspL0jcSTV0trKd/7w6mHVnSxiMK4oh0kluVx0UjmnTBvN3CmjRkxvs5aqz1wz+zAwz92vCoY/AZzm7tclzfNKME9VMPx6MM+ubsu6BrgmGDwaGOiN8aXArj7nyiza5nDQNofD4WzzFHcv62nCiLhY7O53Ance7nLMbIm7zx2EkkYMbXM4aJvDIVXbnMpbJ7YCFUnD5cG4HucxsxhQTOKisYiIDJFUBsFiYIaZTTOzLOBSYGG3eRYCVwSvPww8M2yvD4iIZKiUnRpy93Yzuw54isTto3e7+2ozuxVY4u4LgR8D95tZJbCbRFik0mGfXhqBtM3hoG0Oh5Rsc8ouFouIyMgw/JpXiojIkFIQiIiEXGiCwMzmmdl6M6s0sy+nu55UMLMKM3vWzNaY2WozuyEYP9rMfm9mrwX/DtEjZYaGmUXNbLmZ/V8wPM3MFgX7+hfBzQoZw8xKzOxhM1tnZmvN7IwQ7OMvBH/Tr5jZA2aWk2n72czuNrPqoH1V17ge96sl3B5s+yozO+lw1h2KIEjq7mI+MBO4zMxmpreqlGgHbnL3mcDpwLXBdn4ZeNrdZwBPB8OZ5AZgbdLwfwHfcffpwB4SXZlkku8Bv3X3Y4BZJLY9Y/exmU0CrgfmuvvxJG4+6eqSJpP28z3AvG7jetuv84EZwc81wA8PZ8WhCAKSurtw91agq7uLjOLu2919WfC6gcQHxCQS23pvMNu9wN+mp8LBZ2blwN8AdwXDBryHRJclkHnbWwy8k8Qdd7h7q7vXkcH7OBADcoP2RnnAdjJsP7v78yTunkzW235dANznCS8DJWY2YaDrDksQTAK2JA1XBeMylplNBeYAi4Bx7r49mLQDGJemslLhu8CXgM5geAxQ5+5dDyDItH09DagBfhKcDrvLzPLJ4H3s7luBbwObSQTAXmApmb2fu/S2Xwf1My0sQRAqZlYA/Ar4vLvXJ08LGuxlxD3DZvYBoNrdl6a7liEUA04Cfujuc4BGup0GyqR9DBCcF19AIgQnAvm8/RRKxkvlfg1LEPSnu4uMYGZxEiHwM3d/JBi9s+uwMfi3Ol31DbKzgAvNbBOJ033vIXH+vCQ4hQCZt6+rgCp3XxQMP0wiGDJ1HwO8F9jo7jXu3gY8QmLfZ/J+7tLbfh3Uz7SwBEF/ursY8YLz4z8G1rr7bUmTkrvyuAJ4fKhrSwV3/yd3L3f3qST26TPu/jHgWRJdlkAGbS+Au+8AtphZ10MEziXRtXtG7uPAZuB0M8sL/sa7tjlj93OS3vbrQuDy4O6h04G9SaeQDp27h+IHOB94FXgd+Eq660nRNr6DxKHjKmBF8HM+ifPmTwOvAX8ARqe71hRs+znA/wWvjwD+AlQCvwSy013fIG/rbGBJsJ8fA0Zl+j4GvgasA14B7geyM20/Aw+QuAbSRuLI78re9itgJO6EfB34K4k7qga8bnUxISIScmE5NSQiIr1QEIiIhJyCQEQk5BQEIiIhpyAQEQk5BYGMeGbmZvbTpOGYmdV09UZ6kPfNNrPzk4ZvMbMvHsJ6pyb3FHmQeT6aNDzXzG7v7zoOstxsM/tt0BvnZ5PG33m4PVFK+CgIJBM0AsebWW4w/D7618pyNol2Fqk0FTgQBO6+xN2vH4Tlvh94ETgR+ASAmc0Coh50PCjSXwoCyRRPkuiFFOAyEo1zADCz/KCv978EHbUtCFqY3wpcYmYrzOySYPaZZvZHM9tgZtcH77/VzD6ftLxvWPCsh6RxU83sBTNbFvycGUz6T+DsYB1fMLNz7M3nJow2s8eC/uRfNrMTg/G3BPW+pY5u2kj0whkn0bgI4OvAvw70FyjhpSCQTPEgcKmZ5ZD4lrwoadpXSHQ/cSrwbuBbJD5AbwZ+4e6z3f0XwbzHkPi2fSrw1aDvpruBywHMLEKiO4uf8lbVwPvc/STgEqDr9M+XgReCdXyn23u+Bix39xOBfwbuS5rWUx3Jfk/iaONl4HYzuxBY5u7bDv5rEnm7WN+ziAx/7r4q6Hr7MhJHB8nOI9E5Xdf5/xxgci+LesLdW4AWM6sm0Q3wJjOrNbM5JLoBXu7utWZWmPS+OPB9M5sNdABH9aPsdwAfCup/xszGmFlRb3WQ6Haga3vbCU45BSHxFLDAzG4Ltu0+d8+4/rQkNRQEkkkWkui3/hwSfbR0MeBD7r4+eWYzO62HZbQkve7gzf8jdwGfBMaTOELo7gvAThJPDIsAzYdcff/q6MlnSRxNnE6ir/5LgGfIwI4VJTV0akgyyd3A19z9r93GPwV8Lui5kuCbPUADUEj/PEqiD/xTguV1Vwxsd/dOEhdvo/1YxwvAx4KazgF2ebfnR/Ql6Kv/AySCII/EA3ocyD3Y+0SSKQgkY7h7lbv3dGvm10mculllZquDYUh0Yzyz28Xi3pbdGsz/kLt39DDLD4ArzGwlifP7jcH4VUCHma00sy90e88twMlmtorEReUrOHQ3A98IAugp4GwSvVHeP4BlSUip91GRfgguEi8DPuLur6W7HpHBpCMCkT6Y2UwSfd4/rRCQTKQjAhGRkNMRgYhIyCkIRERCTkEgIhJyCgIRkZBTEIiIhNz/B7BN4bDy5QaBAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Визуализация уровня метилирования."
      ],
      "metadata": {
        "id": "NKAVF62Dck8Y"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Вручную создан файл tracks.ini."
      ],
      "metadata": {
        "id": "0BWdsiAFcBOn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pyGenomeTracks --tracks tracks.ini --region chr11:3100030-3500030 -o image_cov.png"
      ],
      "metadata": {
        "id": "iWZ3EH0XchWR",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "26c7f401-6d1d-406b-8a05-16882435824e"
      },
      "execution_count": 66,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [8Cell-coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [8Cell-methylation]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [ICM-coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [ICM-methylation]\n",
            "INFO:pygenometracks.tracksClass:initialize 5. [Epiblast-coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 6. [Epiblast-methylation]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.018850326538085938\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 12.765957446808512. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [8Cell-coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [8Cell-methylation]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [ICM-coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [ICM-methylation]\n",
            "INFO:pygenometracks.tracksClass:plotting 5. [Epiblast-coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 6. [Epiblast-methylation]\n"
          ]
        }
      ]
    }
  ]
}